// @aws-sdk/client-codedeploy/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var utilWaiter = require('@smithy/util-waiter');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "codedeploy",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class CodeDeployClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultCodeDeployHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let CodeDeployServiceException$1 = class CodeDeployServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, CodeDeployServiceException.prototype);
    }
};

let InstanceLimitExceededException$1 = class InstanceLimitExceededException extends CodeDeployServiceException$1 {
    name = "InstanceLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceLimitExceededException.prototype);
    }
};
let InstanceNameRequiredException$1 = class InstanceNameRequiredException extends CodeDeployServiceException$1 {
    name = "InstanceNameRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceNameRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceNameRequiredException.prototype);
    }
};
let InstanceNotRegisteredException$1 = class InstanceNotRegisteredException extends CodeDeployServiceException$1 {
    name = "InstanceNotRegisteredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceNotRegisteredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceNotRegisteredException.prototype);
    }
};
let InvalidInstanceNameException$1 = class InvalidInstanceNameException extends CodeDeployServiceException$1 {
    name = "InvalidInstanceNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInstanceNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInstanceNameException.prototype);
    }
};
let InvalidTagException$1 = class InvalidTagException extends CodeDeployServiceException$1 {
    name = "InvalidTagException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTagException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTagException.prototype);
    }
};
let TagLimitExceededException$1 = class TagLimitExceededException extends CodeDeployServiceException$1 {
    name = "TagLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TagLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagLimitExceededException.prototype);
    }
};
let TagRequiredException$1 = class TagRequiredException extends CodeDeployServiceException$1 {
    name = "TagRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TagRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagRequiredException.prototype);
    }
};
let AlarmsLimitExceededException$1 = class AlarmsLimitExceededException extends CodeDeployServiceException$1 {
    name = "AlarmsLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AlarmsLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AlarmsLimitExceededException.prototype);
    }
};
let ApplicationAlreadyExistsException$1 = class ApplicationAlreadyExistsException extends CodeDeployServiceException$1 {
    name = "ApplicationAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApplicationAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApplicationAlreadyExistsException.prototype);
    }
};
let ApplicationDoesNotExistException$1 = class ApplicationDoesNotExistException extends CodeDeployServiceException$1 {
    name = "ApplicationDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApplicationDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApplicationDoesNotExistException.prototype);
    }
};
const ComputePlatform = {
    ECS: "ECS",
    LAMBDA: "Lambda",
    SERVER: "Server",
};
let ApplicationLimitExceededException$1 = class ApplicationLimitExceededException extends CodeDeployServiceException$1 {
    name = "ApplicationLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApplicationLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApplicationLimitExceededException.prototype);
    }
};
let ApplicationNameRequiredException$1 = class ApplicationNameRequiredException extends CodeDeployServiceException$1 {
    name = "ApplicationNameRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ApplicationNameRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ApplicationNameRequiredException.prototype);
    }
};
const ApplicationRevisionSortBy = {
    FirstUsedTime: "firstUsedTime",
    LastUsedTime: "lastUsedTime",
    RegisterTime: "registerTime",
};
let ArnNotSupportedException$1 = class ArnNotSupportedException extends CodeDeployServiceException$1 {
    name = "ArnNotSupportedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ArnNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ArnNotSupportedException.prototype);
    }
};
const AutoRollbackEvent = {
    DEPLOYMENT_FAILURE: "DEPLOYMENT_FAILURE",
    DEPLOYMENT_STOP_ON_ALARM: "DEPLOYMENT_STOP_ON_ALARM",
    DEPLOYMENT_STOP_ON_REQUEST: "DEPLOYMENT_STOP_ON_REQUEST",
};
const RevisionLocationType = {
    AppSpecContent: "AppSpecContent",
    GitHub: "GitHub",
    S3: "S3",
    String: "String",
};
const BundleType = {
    JSON: "JSON",
    Tar: "tar",
    TarGZip: "tgz",
    YAML: "YAML",
    Zip: "zip",
};
let BatchLimitExceededException$1 = class BatchLimitExceededException extends CodeDeployServiceException$1 {
    name = "BatchLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BatchLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchLimitExceededException.prototype);
    }
};
let InvalidApplicationNameException$1 = class InvalidApplicationNameException extends CodeDeployServiceException$1 {
    name = "InvalidApplicationNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidApplicationNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidApplicationNameException.prototype);
    }
};
let InvalidRevisionException$1 = class InvalidRevisionException extends CodeDeployServiceException$1 {
    name = "InvalidRevisionException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidRevisionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRevisionException.prototype);
    }
};
let RevisionRequiredException$1 = class RevisionRequiredException extends CodeDeployServiceException$1 {
    name = "RevisionRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RevisionRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RevisionRequiredException.prototype);
    }
};
const DeploymentReadyAction = {
    CONTINUE_DEPLOYMENT: "CONTINUE_DEPLOYMENT",
    STOP_DEPLOYMENT: "STOP_DEPLOYMENT",
};
const GreenFleetProvisioningAction = {
    COPY_AUTO_SCALING_GROUP: "COPY_AUTO_SCALING_GROUP",
    DISCOVER_EXISTING: "DISCOVER_EXISTING",
};
const InstanceAction = {
    KEEP_ALIVE: "KEEP_ALIVE",
    TERMINATE: "TERMINATE",
};
const DeploymentOption = {
    WITHOUT_TRAFFIC_CONTROL: "WITHOUT_TRAFFIC_CONTROL",
    WITH_TRAFFIC_CONTROL: "WITH_TRAFFIC_CONTROL",
};
const DeploymentType = {
    BLUE_GREEN: "BLUE_GREEN",
    IN_PLACE: "IN_PLACE",
};
const EC2TagFilterType = {
    KEY_AND_VALUE: "KEY_AND_VALUE",
    KEY_ONLY: "KEY_ONLY",
    VALUE_ONLY: "VALUE_ONLY",
};
const DeploymentStatus = {
    BAKING: "Baking",
    CREATED: "Created",
    FAILED: "Failed",
    IN_PROGRESS: "InProgress",
    QUEUED: "Queued",
    READY: "Ready",
    STOPPED: "Stopped",
    SUCCEEDED: "Succeeded",
};
const TagFilterType = {
    KEY_AND_VALUE: "KEY_AND_VALUE",
    KEY_ONLY: "KEY_ONLY",
    VALUE_ONLY: "VALUE_ONLY",
};
const OutdatedInstancesStrategy = {
    Ignore: "IGNORE",
    Update: "UPDATE",
};
const TriggerEventType = {
    DEPLOYMENT_FAILURE: "DeploymentFailure",
    DEPLOYMENT_READY: "DeploymentReady",
    DEPLOYMENT_ROLLBACK: "DeploymentRollback",
    DEPLOYMENT_START: "DeploymentStart",
    DEPLOYMENT_STOP: "DeploymentStop",
    DEPLOYMENT_SUCCESS: "DeploymentSuccess",
    INSTANCE_FAILURE: "InstanceFailure",
    INSTANCE_READY: "InstanceReady",
    INSTANCE_START: "InstanceStart",
    INSTANCE_SUCCESS: "InstanceSuccess",
};
let DeploymentConfigDoesNotExistException$1 = class DeploymentConfigDoesNotExistException extends CodeDeployServiceException$1 {
    name = "DeploymentConfigDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentConfigDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentConfigDoesNotExistException.prototype);
    }
};
let DeploymentGroupNameRequiredException$1 = class DeploymentGroupNameRequiredException extends CodeDeployServiceException$1 {
    name = "DeploymentGroupNameRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentGroupNameRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentGroupNameRequiredException.prototype);
    }
};
let InvalidDeploymentGroupNameException$1 = class InvalidDeploymentGroupNameException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentGroupNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentGroupNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentGroupNameException.prototype);
    }
};
const _InstanceType = {
    BLUE: "Blue",
    GREEN: "Green",
};
const LifecycleErrorCode = {
    SCRIPT_FAILED: "ScriptFailed",
    SCRIPT_MISSING: "ScriptMissing",
    SCRIPT_NOT_EXECUTABLE: "ScriptNotExecutable",
    SCRIPT_TIMED_OUT: "ScriptTimedOut",
    SUCCESS: "Success",
    UNKNOWN_ERROR: "UnknownError",
};
const LifecycleEventStatus = {
    FAILED: "Failed",
    IN_PROGRESS: "InProgress",
    PENDING: "Pending",
    SKIPPED: "Skipped",
    SUCCEEDED: "Succeeded",
    UNKNOWN: "Unknown",
};
const InstanceStatus = {
    FAILED: "Failed",
    IN_PROGRESS: "InProgress",
    PENDING: "Pending",
    READY: "Ready",
    SKIPPED: "Skipped",
    SUCCEEDED: "Succeeded",
    UNKNOWN: "Unknown",
};
let DeploymentDoesNotExistException$1 = class DeploymentDoesNotExistException extends CodeDeployServiceException$1 {
    name = "DeploymentDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentDoesNotExistException.prototype);
    }
};
let DeploymentIdRequiredException$1 = class DeploymentIdRequiredException extends CodeDeployServiceException$1 {
    name = "DeploymentIdRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentIdRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentIdRequiredException.prototype);
    }
};
let InstanceIdRequiredException$1 = class InstanceIdRequiredException extends CodeDeployServiceException$1 {
    name = "InstanceIdRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceIdRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceIdRequiredException.prototype);
    }
};
let InvalidComputePlatformException$1 = class InvalidComputePlatformException extends CodeDeployServiceException$1 {
    name = "InvalidComputePlatformException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidComputePlatformException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidComputePlatformException.prototype);
    }
};
let InvalidDeploymentIdException$1 = class InvalidDeploymentIdException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentIdException.prototype);
    }
};
const DeploymentCreator = {
    Autoscaling: "autoscaling",
    AutoscalingTermination: "autoscalingTermination",
    CloudFormation: "CloudFormation",
    CloudFormationRollback: "CloudFormationRollback",
    CodeDeploy: "CodeDeploy",
    CodeDeployAutoUpdate: "CodeDeployAutoUpdate",
    CodeDeployRollback: "codeDeployRollback",
    User: "user",
};
const ErrorCode = {
    AGENT_ISSUE: "AGENT_ISSUE",
    ALARM_ACTIVE: "ALARM_ACTIVE",
    APPLICATION_MISSING: "APPLICATION_MISSING",
    AUTOSCALING_VALIDATION_ERROR: "AUTOSCALING_VALIDATION_ERROR",
    AUTO_SCALING_CONFIGURATION: "AUTO_SCALING_CONFIGURATION",
    AUTO_SCALING_IAM_ROLE_PERMISSIONS: "AUTO_SCALING_IAM_ROLE_PERMISSIONS",
    CLOUDFORMATION_STACK_FAILURE: "CLOUDFORMATION_STACK_FAILURE",
    CODEDEPLOY_RESOURCE_CANNOT_BE_FOUND: "CODEDEPLOY_RESOURCE_CANNOT_BE_FOUND",
    CUSTOMER_APPLICATION_UNHEALTHY: "CUSTOMER_APPLICATION_UNHEALTHY",
    DEPLOYMENT_GROUP_MISSING: "DEPLOYMENT_GROUP_MISSING",
    ECS_UPDATE_ERROR: "ECS_UPDATE_ERROR",
    ELASTIC_LOAD_BALANCING_INVALID: "ELASTIC_LOAD_BALANCING_INVALID",
    ELB_INVALID_INSTANCE: "ELB_INVALID_INSTANCE",
    HEALTH_CONSTRAINTS: "HEALTH_CONSTRAINTS",
    HEALTH_CONSTRAINTS_INVALID: "HEALTH_CONSTRAINTS_INVALID",
    HOOK_EXECUTION_FAILURE: "HOOK_EXECUTION_FAILURE",
    IAM_ROLE_MISSING: "IAM_ROLE_MISSING",
    IAM_ROLE_PERMISSIONS: "IAM_ROLE_PERMISSIONS",
    INTERNAL_ERROR: "INTERNAL_ERROR",
    INVALID_ECS_SERVICE: "INVALID_ECS_SERVICE",
    INVALID_LAMBDA_CONFIGURATION: "INVALID_LAMBDA_CONFIGURATION",
    INVALID_LAMBDA_FUNCTION: "INVALID_LAMBDA_FUNCTION",
    INVALID_REVISION: "INVALID_REVISION",
    MANUAL_STOP: "MANUAL_STOP",
    MISSING_BLUE_GREEN_DEPLOYMENT_CONFIGURATION: "MISSING_BLUE_GREEN_DEPLOYMENT_CONFIGURATION",
    MISSING_ELB_INFORMATION: "MISSING_ELB_INFORMATION",
    MISSING_GITHUB_TOKEN: "MISSING_GITHUB_TOKEN",
    NO_EC2_SUBSCRIPTION: "NO_EC2_SUBSCRIPTION",
    NO_INSTANCES: "NO_INSTANCES",
    OVER_MAX_INSTANCES: "OVER_MAX_INSTANCES",
    RESOURCE_LIMIT_EXCEEDED: "RESOURCE_LIMIT_EXCEEDED",
    REVISION_MISSING: "REVISION_MISSING",
    THROTTLED: "THROTTLED",
    TIMEOUT: "TIMEOUT",
};
const FileExistsBehavior = {
    DISALLOW: "DISALLOW",
    OVERWRITE: "OVERWRITE",
    RETAIN: "RETAIN",
};
const TargetStatus = {
    FAILED: "Failed",
    IN_PROGRESS: "InProgress",
    PENDING: "Pending",
    READY: "Ready",
    SKIPPED: "Skipped",
    SUCCEEDED: "Succeeded",
    UNKNOWN: "Unknown",
};
const DeploymentTargetType = {
    CLOUDFORMATION_TARGET: "CloudFormationTarget",
    ECS_TARGET: "ECSTarget",
    INSTANCE_TARGET: "InstanceTarget",
    LAMBDA_TARGET: "LambdaTarget",
};
const TargetLabel = {
    BLUE: "Blue",
    GREEN: "Green",
};
let DeploymentNotStartedException$1 = class DeploymentNotStartedException extends CodeDeployServiceException$1 {
    name = "DeploymentNotStartedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentNotStartedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentNotStartedException.prototype);
    }
};
let DeploymentTargetDoesNotExistException$1 = class DeploymentTargetDoesNotExistException extends CodeDeployServiceException$1 {
    name = "DeploymentTargetDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentTargetDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentTargetDoesNotExistException.prototype);
    }
};
let DeploymentTargetIdRequiredException$1 = class DeploymentTargetIdRequiredException extends CodeDeployServiceException$1 {
    name = "DeploymentTargetIdRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentTargetIdRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentTargetIdRequiredException.prototype);
    }
};
let DeploymentTargetListSizeExceededException$1 = class DeploymentTargetListSizeExceededException extends CodeDeployServiceException$1 {
    name = "DeploymentTargetListSizeExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentTargetListSizeExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentTargetListSizeExceededException.prototype);
    }
};
let InstanceDoesNotExistException$1 = class InstanceDoesNotExistException extends CodeDeployServiceException$1 {
    name = "InstanceDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceDoesNotExistException.prototype);
    }
};
let InvalidDeploymentTargetIdException$1 = class InvalidDeploymentTargetIdException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentTargetIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentTargetIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentTargetIdException.prototype);
    }
};
let BucketNameFilterRequiredException$1 = class BucketNameFilterRequiredException extends CodeDeployServiceException$1 {
    name = "BucketNameFilterRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BucketNameFilterRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BucketNameFilterRequiredException.prototype);
    }
};
const DeploymentWaitType = {
    READY_WAIT: "READY_WAIT",
    TERMINATION_WAIT: "TERMINATION_WAIT",
};
let DeploymentAlreadyCompletedException$1 = class DeploymentAlreadyCompletedException extends CodeDeployServiceException$1 {
    name = "DeploymentAlreadyCompletedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentAlreadyCompletedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentAlreadyCompletedException.prototype);
    }
};
let DeploymentIsNotInReadyStateException$1 = class DeploymentIsNotInReadyStateException extends CodeDeployServiceException$1 {
    name = "DeploymentIsNotInReadyStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentIsNotInReadyStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentIsNotInReadyStateException.prototype);
    }
};
let InvalidDeploymentStatusException$1 = class InvalidDeploymentStatusException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentStatusException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentStatusException.prototype);
    }
};
let InvalidDeploymentWaitTypeException$1 = class InvalidDeploymentWaitTypeException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentWaitTypeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentWaitTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentWaitTypeException.prototype);
    }
};
let UnsupportedActionForDeploymentTypeException$1 = class UnsupportedActionForDeploymentTypeException extends CodeDeployServiceException$1 {
    name = "UnsupportedActionForDeploymentTypeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedActionForDeploymentTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedActionForDeploymentTypeException.prototype);
    }
};
let InvalidTagsToAddException$1 = class InvalidTagsToAddException extends CodeDeployServiceException$1 {
    name = "InvalidTagsToAddException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTagsToAddException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTagsToAddException.prototype);
    }
};
let DeploymentGroupDoesNotExistException$1 = class DeploymentGroupDoesNotExistException extends CodeDeployServiceException$1 {
    name = "DeploymentGroupDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentGroupDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentGroupDoesNotExistException.prototype);
    }
};
let DeploymentLimitExceededException$1 = class DeploymentLimitExceededException extends CodeDeployServiceException$1 {
    name = "DeploymentLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentLimitExceededException.prototype);
    }
};
let DescriptionTooLongException$1 = class DescriptionTooLongException extends CodeDeployServiceException$1 {
    name = "DescriptionTooLongException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DescriptionTooLongException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DescriptionTooLongException.prototype);
    }
};
let InvalidAlarmConfigException$1 = class InvalidAlarmConfigException extends CodeDeployServiceException$1 {
    name = "InvalidAlarmConfigException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAlarmConfigException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAlarmConfigException.prototype);
    }
};
let InvalidAutoRollbackConfigException$1 = class InvalidAutoRollbackConfigException extends CodeDeployServiceException$1 {
    name = "InvalidAutoRollbackConfigException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAutoRollbackConfigException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAutoRollbackConfigException.prototype);
    }
};
let InvalidAutoScalingGroupException$1 = class InvalidAutoScalingGroupException extends CodeDeployServiceException$1 {
    name = "InvalidAutoScalingGroupException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAutoScalingGroupException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAutoScalingGroupException.prototype);
    }
};
let InvalidDeploymentConfigNameException$1 = class InvalidDeploymentConfigNameException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentConfigNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentConfigNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentConfigNameException.prototype);
    }
};
let InvalidFileExistsBehaviorException$1 = class InvalidFileExistsBehaviorException extends CodeDeployServiceException$1 {
    name = "InvalidFileExistsBehaviorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidFileExistsBehaviorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidFileExistsBehaviorException.prototype);
    }
};
let InvalidGitHubAccountTokenException$1 = class InvalidGitHubAccountTokenException extends CodeDeployServiceException$1 {
    name = "InvalidGitHubAccountTokenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidGitHubAccountTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidGitHubAccountTokenException.prototype);
    }
};
let InvalidIgnoreApplicationStopFailuresValueException$1 = class InvalidIgnoreApplicationStopFailuresValueException extends CodeDeployServiceException$1 {
    name = "InvalidIgnoreApplicationStopFailuresValueException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidIgnoreApplicationStopFailuresValueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIgnoreApplicationStopFailuresValueException.prototype);
    }
};
let InvalidLoadBalancerInfoException$1 = class InvalidLoadBalancerInfoException extends CodeDeployServiceException$1 {
    name = "InvalidLoadBalancerInfoException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidLoadBalancerInfoException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLoadBalancerInfoException.prototype);
    }
};
let InvalidRoleException$1 = class InvalidRoleException extends CodeDeployServiceException$1 {
    name = "InvalidRoleException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidRoleException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRoleException.prototype);
    }
};
let InvalidTargetInstancesException$1 = class InvalidTargetInstancesException extends CodeDeployServiceException$1 {
    name = "InvalidTargetInstancesException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTargetInstancesException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTargetInstancesException.prototype);
    }
};
let InvalidTrafficRoutingConfigurationException$1 = class InvalidTrafficRoutingConfigurationException extends CodeDeployServiceException$1 {
    name = "InvalidTrafficRoutingConfigurationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTrafficRoutingConfigurationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTrafficRoutingConfigurationException.prototype);
    }
};
let InvalidUpdateOutdatedInstancesOnlyValueException$1 = class InvalidUpdateOutdatedInstancesOnlyValueException extends CodeDeployServiceException$1 {
    name = "InvalidUpdateOutdatedInstancesOnlyValueException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidUpdateOutdatedInstancesOnlyValueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidUpdateOutdatedInstancesOnlyValueException.prototype);
    }
};
let RevisionDoesNotExistException$1 = class RevisionDoesNotExistException extends CodeDeployServiceException$1 {
    name = "RevisionDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RevisionDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RevisionDoesNotExistException.prototype);
    }
};
let ThrottlingException$1 = class ThrottlingException extends CodeDeployServiceException$1 {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
};
const MinimumHealthyHostsType = {
    FLEET_PERCENT: "FLEET_PERCENT",
    HOST_COUNT: "HOST_COUNT",
};
const TrafficRoutingType = {
    AllAtOnce: "AllAtOnce",
    TimeBasedCanary: "TimeBasedCanary",
    TimeBasedLinear: "TimeBasedLinear",
};
const MinimumHealthyHostsPerZoneType = {
    FLEET_PERCENT: "FLEET_PERCENT",
    HOST_COUNT: "HOST_COUNT",
};
let DeploymentConfigAlreadyExistsException$1 = class DeploymentConfigAlreadyExistsException extends CodeDeployServiceException$1 {
    name = "DeploymentConfigAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentConfigAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentConfigAlreadyExistsException.prototype);
    }
};
let DeploymentConfigLimitExceededException$1 = class DeploymentConfigLimitExceededException extends CodeDeployServiceException$1 {
    name = "DeploymentConfigLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentConfigLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentConfigLimitExceededException.prototype);
    }
};
let DeploymentConfigNameRequiredException$1 = class DeploymentConfigNameRequiredException extends CodeDeployServiceException$1 {
    name = "DeploymentConfigNameRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentConfigNameRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentConfigNameRequiredException.prototype);
    }
};
let InvalidMinimumHealthyHostValueException$1 = class InvalidMinimumHealthyHostValueException extends CodeDeployServiceException$1 {
    name = "InvalidMinimumHealthyHostValueException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidMinimumHealthyHostValueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidMinimumHealthyHostValueException.prototype);
    }
};
let InvalidZonalDeploymentConfigurationException$1 = class InvalidZonalDeploymentConfigurationException extends CodeDeployServiceException$1 {
    name = "InvalidZonalDeploymentConfigurationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidZonalDeploymentConfigurationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidZonalDeploymentConfigurationException.prototype);
    }
};
let DeploymentGroupAlreadyExistsException$1 = class DeploymentGroupAlreadyExistsException extends CodeDeployServiceException$1 {
    name = "DeploymentGroupAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentGroupAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentGroupAlreadyExistsException.prototype);
    }
};
let DeploymentGroupLimitExceededException$1 = class DeploymentGroupLimitExceededException extends CodeDeployServiceException$1 {
    name = "DeploymentGroupLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentGroupLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentGroupLimitExceededException.prototype);
    }
};
let ECSServiceMappingLimitExceededException$1 = class ECSServiceMappingLimitExceededException extends CodeDeployServiceException$1 {
    name = "ECSServiceMappingLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ECSServiceMappingLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ECSServiceMappingLimitExceededException.prototype);
    }
};
let InvalidBlueGreenDeploymentConfigurationException$1 = class InvalidBlueGreenDeploymentConfigurationException extends CodeDeployServiceException$1 {
    name = "InvalidBlueGreenDeploymentConfigurationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidBlueGreenDeploymentConfigurationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidBlueGreenDeploymentConfigurationException.prototype);
    }
};
let InvalidDeploymentStyleException$1 = class InvalidDeploymentStyleException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentStyleException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentStyleException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentStyleException.prototype);
    }
};
let InvalidEC2TagCombinationException$1 = class InvalidEC2TagCombinationException extends CodeDeployServiceException$1 {
    name = "InvalidEC2TagCombinationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidEC2TagCombinationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidEC2TagCombinationException.prototype);
    }
};
let InvalidEC2TagException$1 = class InvalidEC2TagException extends CodeDeployServiceException$1 {
    name = "InvalidEC2TagException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidEC2TagException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidEC2TagException.prototype);
    }
};
let InvalidECSServiceException$1 = class InvalidECSServiceException extends CodeDeployServiceException$1 {
    name = "InvalidECSServiceException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidECSServiceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidECSServiceException.prototype);
    }
};
let InvalidInputException$1 = class InvalidInputException extends CodeDeployServiceException$1 {
    name = "InvalidInputException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInputException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInputException.prototype);
    }
};
let InvalidOnPremisesTagCombinationException$1 = class InvalidOnPremisesTagCombinationException extends CodeDeployServiceException$1 {
    name = "InvalidOnPremisesTagCombinationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidOnPremisesTagCombinationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOnPremisesTagCombinationException.prototype);
    }
};
let InvalidTargetGroupPairException$1 = class InvalidTargetGroupPairException extends CodeDeployServiceException$1 {
    name = "InvalidTargetGroupPairException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTargetGroupPairException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTargetGroupPairException.prototype);
    }
};
let InvalidTriggerConfigException$1 = class InvalidTriggerConfigException extends CodeDeployServiceException$1 {
    name = "InvalidTriggerConfigException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTriggerConfigException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTriggerConfigException.prototype);
    }
};
let LifecycleHookLimitExceededException$1 = class LifecycleHookLimitExceededException extends CodeDeployServiceException$1 {
    name = "LifecycleHookLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LifecycleHookLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LifecycleHookLimitExceededException.prototype);
    }
};
let RoleRequiredException$1 = class RoleRequiredException extends CodeDeployServiceException$1 {
    name = "RoleRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RoleRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RoleRequiredException.prototype);
    }
};
let TagSetListLimitExceededException$1 = class TagSetListLimitExceededException extends CodeDeployServiceException$1 {
    name = "TagSetListLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TagSetListLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagSetListLimitExceededException.prototype);
    }
};
let TriggerTargetsLimitExceededException$1 = class TriggerTargetsLimitExceededException extends CodeDeployServiceException$1 {
    name = "TriggerTargetsLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TriggerTargetsLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TriggerTargetsLimitExceededException.prototype);
    }
};
let DeploymentConfigInUseException$1 = class DeploymentConfigInUseException extends CodeDeployServiceException$1 {
    name = "DeploymentConfigInUseException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeploymentConfigInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeploymentConfigInUseException.prototype);
    }
};
let InvalidOperationException$1 = class InvalidOperationException extends CodeDeployServiceException$1 {
    name = "InvalidOperationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidOperationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOperationException.prototype);
    }
};
let GitHubAccountTokenDoesNotExistException$1 = class GitHubAccountTokenDoesNotExistException extends CodeDeployServiceException$1 {
    name = "GitHubAccountTokenDoesNotExistException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "GitHubAccountTokenDoesNotExistException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GitHubAccountTokenDoesNotExistException.prototype);
    }
};
let GitHubAccountTokenNameRequiredException$1 = class GitHubAccountTokenNameRequiredException extends CodeDeployServiceException$1 {
    name = "GitHubAccountTokenNameRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "GitHubAccountTokenNameRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GitHubAccountTokenNameRequiredException.prototype);
    }
};
let InvalidGitHubAccountTokenNameException$1 = class InvalidGitHubAccountTokenNameException extends CodeDeployServiceException$1 {
    name = "InvalidGitHubAccountTokenNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidGitHubAccountTokenNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidGitHubAccountTokenNameException.prototype);
    }
};
let OperationNotSupportedException$1 = class OperationNotSupportedException extends CodeDeployServiceException$1 {
    name = "OperationNotSupportedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "OperationNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationNotSupportedException.prototype);
    }
};
let ResourceValidationException$1 = class ResourceValidationException extends CodeDeployServiceException$1 {
    name = "ResourceValidationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceValidationException.prototype);
    }
};
let InvalidBucketNameFilterException$1 = class InvalidBucketNameFilterException extends CodeDeployServiceException$1 {
    name = "InvalidBucketNameFilterException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidBucketNameFilterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidBucketNameFilterException.prototype);
    }
};
let InvalidDeployedStateFilterException$1 = class InvalidDeployedStateFilterException extends CodeDeployServiceException$1 {
    name = "InvalidDeployedStateFilterException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeployedStateFilterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeployedStateFilterException.prototype);
    }
};
let InvalidKeyPrefixFilterException$1 = class InvalidKeyPrefixFilterException extends CodeDeployServiceException$1 {
    name = "InvalidKeyPrefixFilterException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKeyPrefixFilterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKeyPrefixFilterException.prototype);
    }
};
let InvalidNextTokenException$1 = class InvalidNextTokenException extends CodeDeployServiceException$1 {
    name = "InvalidNextTokenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidNextTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidNextTokenException.prototype);
    }
};
let InvalidSortByException$1 = class InvalidSortByException extends CodeDeployServiceException$1 {
    name = "InvalidSortByException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSortByException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSortByException.prototype);
    }
};
let InvalidSortOrderException$1 = class InvalidSortOrderException extends CodeDeployServiceException$1 {
    name = "InvalidSortOrderException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSortOrderException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSortOrderException.prototype);
    }
};
const ListStateFilterAction = {
    Exclude: "exclude",
    Ignore: "ignore",
    Include: "include",
};
const SortOrder = {
    Ascending: "ascending",
    Descending: "descending",
};
let InvalidDeploymentInstanceTypeException$1 = class InvalidDeploymentInstanceTypeException extends CodeDeployServiceException$1 {
    name = "InvalidDeploymentInstanceTypeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDeploymentInstanceTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDeploymentInstanceTypeException.prototype);
    }
};
let InvalidInstanceStatusException$1 = class InvalidInstanceStatusException extends CodeDeployServiceException$1 {
    name = "InvalidInstanceStatusException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInstanceStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInstanceStatusException.prototype);
    }
};
let InvalidInstanceTypeException$1 = class InvalidInstanceTypeException extends CodeDeployServiceException$1 {
    name = "InvalidInstanceTypeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInstanceTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInstanceTypeException.prototype);
    }
};
let InvalidTargetFilterNameException$1 = class InvalidTargetFilterNameException extends CodeDeployServiceException$1 {
    name = "InvalidTargetFilterNameException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTargetFilterNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTargetFilterNameException.prototype);
    }
};
let InvalidExternalIdException$1 = class InvalidExternalIdException extends CodeDeployServiceException$1 {
    name = "InvalidExternalIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidExternalIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidExternalIdException.prototype);
    }
};
let InvalidTimeRangeException$1 = class InvalidTimeRangeException extends CodeDeployServiceException$1 {
    name = "InvalidTimeRangeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTimeRangeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTimeRangeException.prototype);
    }
};
const TargetFilterName = {
    SERVER_INSTANCE_LABEL: "ServerInstanceLabel",
    TARGET_STATUS: "TargetStatus",
};
let InvalidRegistrationStatusException$1 = class InvalidRegistrationStatusException extends CodeDeployServiceException$1 {
    name = "InvalidRegistrationStatusException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidRegistrationStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRegistrationStatusException.prototype);
    }
};
let InvalidTagFilterException$1 = class InvalidTagFilterException extends CodeDeployServiceException$1 {
    name = "InvalidTagFilterException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTagFilterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTagFilterException.prototype);
    }
};
const RegistrationStatus = {
    Deregistered: "Deregistered",
    Registered: "Registered",
};
let InvalidArnException$1 = class InvalidArnException extends CodeDeployServiceException$1 {
    name = "InvalidArnException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidArnException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidArnException.prototype);
    }
};
let ResourceArnRequiredException$1 = class ResourceArnRequiredException extends CodeDeployServiceException$1 {
    name = "ResourceArnRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceArnRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceArnRequiredException.prototype);
    }
};
let InvalidLifecycleEventHookExecutionIdException$1 = class InvalidLifecycleEventHookExecutionIdException extends CodeDeployServiceException$1 {
    name = "InvalidLifecycleEventHookExecutionIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidLifecycleEventHookExecutionIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLifecycleEventHookExecutionIdException.prototype);
    }
};
let InvalidLifecycleEventHookExecutionStatusException$1 = class InvalidLifecycleEventHookExecutionStatusException extends CodeDeployServiceException$1 {
    name = "InvalidLifecycleEventHookExecutionStatusException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidLifecycleEventHookExecutionStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLifecycleEventHookExecutionStatusException.prototype);
    }
};
let LifecycleEventAlreadyCompletedException$1 = class LifecycleEventAlreadyCompletedException extends CodeDeployServiceException$1 {
    name = "LifecycleEventAlreadyCompletedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LifecycleEventAlreadyCompletedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LifecycleEventAlreadyCompletedException.prototype);
    }
};
let IamArnRequiredException$1 = class IamArnRequiredException extends CodeDeployServiceException$1 {
    name = "IamArnRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IamArnRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IamArnRequiredException.prototype);
    }
};
let IamSessionArnAlreadyRegisteredException$1 = class IamSessionArnAlreadyRegisteredException extends CodeDeployServiceException$1 {
    name = "IamSessionArnAlreadyRegisteredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IamSessionArnAlreadyRegisteredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IamSessionArnAlreadyRegisteredException.prototype);
    }
};
let IamUserArnAlreadyRegisteredException$1 = class IamUserArnAlreadyRegisteredException extends CodeDeployServiceException$1 {
    name = "IamUserArnAlreadyRegisteredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IamUserArnAlreadyRegisteredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IamUserArnAlreadyRegisteredException.prototype);
    }
};
let IamUserArnRequiredException$1 = class IamUserArnRequiredException extends CodeDeployServiceException$1 {
    name = "IamUserArnRequiredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IamUserArnRequiredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IamUserArnRequiredException.prototype);
    }
};
let InstanceNameAlreadyRegisteredException$1 = class InstanceNameAlreadyRegisteredException extends CodeDeployServiceException$1 {
    name = "InstanceNameAlreadyRegisteredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InstanceNameAlreadyRegisteredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InstanceNameAlreadyRegisteredException.prototype);
    }
};
let InvalidIamSessionArnException$1 = class InvalidIamSessionArnException extends CodeDeployServiceException$1 {
    name = "InvalidIamSessionArnException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidIamSessionArnException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIamSessionArnException.prototype);
    }
};
let InvalidIamUserArnException$1 = class InvalidIamUserArnException extends CodeDeployServiceException$1 {
    name = "InvalidIamUserArnException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidIamUserArnException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIamUserArnException.prototype);
    }
};
let MultipleIamArnsProvidedException$1 = class MultipleIamArnsProvidedException extends CodeDeployServiceException$1 {
    name = "MultipleIamArnsProvidedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MultipleIamArnsProvidedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MultipleIamArnsProvidedException.prototype);
    }
};
const StopStatus = {
    PENDING: "Pending",
    SUCCEEDED: "Succeeded",
};

const _A = "Alarm";
const _AAEE = "ApplicationAlreadyExistsException";
const _AC = "AlarmConfiguration";
const _ADNEE = "ApplicationDoesNotExistException";
const _AI = "ApplicationInfo";
const _AIL = "ApplicationsInfoList";
const _AL = "AlarmList";
const _ALEE = "AlarmsLimitExceededException";
const _ALEEp = "ApplicationLimitExceededException";
const _ANRE = "ApplicationNameRequiredException";
const _ANSE = "ArnNotSupportedException";
const _ARC = "AutoRollbackConfiguration";
const _ASC = "AppSpecContent";
const _ASG = "AutoScalingGroup";
const _ASGL = "AutoScalingGroupList";
const _ATTOPI = "AddTagsToOnPremisesInstances";
const _ATTOPII = "AddTagsToOnPremisesInstancesInput";
const _BGA = "BatchGetApplications";
const _BGAI = "BatchGetApplicationsInput";
const _BGAO = "BatchGetApplicationsOutput";
const _BGAR = "BatchGetApplicationRevisions";
const _BGARI = "BatchGetApplicationRevisionsInput";
const _BGARO = "BatchGetApplicationRevisionsOutput";
const _BGD = "BatchGetDeployments";
const _BGDC = "BlueGreenDeploymentConfiguration";
const _BGDG = "BatchGetDeploymentGroups";
const _BGDGI = "BatchGetDeploymentGroupsInput";
const _BGDGO = "BatchGetDeploymentGroupsOutput";
const _BGDI = "BatchGetDeploymentsInput";
const _BGDII = "BatchGetDeploymentInstancesInput";
const _BGDIO = "BatchGetDeploymentInstancesOutput";
const _BGDIa = "BatchGetDeploymentInstances";
const _BGDO = "BatchGetDeploymentsOutput";
const _BGDT = "BatchGetDeploymentTargets";
const _BGDTI = "BatchGetDeploymentTargetsInput";
const _BGDTO = "BatchGetDeploymentTargetsOutput";
const _BGOPI = "BatchGetOnPremisesInstances";
const _BGOPII = "BatchGetOnPremisesInstancesInput";
const _BGOPIO = "BatchGetOnPremisesInstancesOutput";
const _BITO = "BlueInstanceTerminationOption";
const _BLEE = "BatchLimitExceededException";
const _BNFRE = "BucketNameFilterRequiredException";
const _CA = "CreateApplication";
const _CAI = "CreateApplicationInput";
const _CAO = "CreateApplicationOutput";
const _CD = "ContinueDeployment";
const _CDC = "CreateDeploymentConfig";
const _CDCI = "CreateDeploymentConfigInput";
const _CDCO = "CreateDeploymentConfigOutput";
const _CDG = "CreateDeploymentGroup";
const _CDGI = "CreateDeploymentGroupInput";
const _CDGO = "CreateDeploymentGroupOutput";
const _CDI = "ContinueDeploymentInput";
const _CDIr = "CreateDeploymentInput";
const _CDO = "CreateDeploymentOutput";
const _CDr = "CreateDeployment";
const _CFT = "CloudFormationTarget";
const _D = "Diagnostics";
const _DA = "DeleteApplication";
const _DACE = "DeploymentAlreadyCompletedException";
const _DAI = "DeleteApplicationInput";
const _DCAEE = "DeploymentConfigAlreadyExistsException";
const _DCDNEE = "DeploymentConfigDoesNotExistException";
const _DCI = "DeploymentConfigInfo";
const _DCIUE = "DeploymentConfigInUseException";
const _DCLEE = "DeploymentConfigLimitExceededException";
const _DCNRE = "DeploymentConfigNameRequiredException";
const _DDC = "DeleteDeploymentConfig";
const _DDCI = "DeleteDeploymentConfigInput";
const _DDG = "DeleteDeploymentGroup";
const _DDGI = "DeleteDeploymentGroupInput";
const _DDGO = "DeleteDeploymentGroupOutput";
const _DDNEE = "DeploymentDoesNotExistException";
const _DGAEE = "DeploymentGroupAlreadyExistsException";
const _DGDNEE = "DeploymentGroupDoesNotExistException";
const _DGHAT = "DeleteGitHubAccountToken";
const _DGHATI = "DeleteGitHubAccountTokenInput";
const _DGHATO = "DeleteGitHubAccountTokenOutput";
const _DGI = "DeploymentGroupInfo";
const _DGIL = "DeploymentGroupInfoList";
const _DGLEE = "DeploymentGroupLimitExceededException";
const _DGNRE = "DeploymentGroupNameRequiredException";
const _DI = "DeploymentInfo";
const _DIL = "DeploymentsInfoList";
const _DINIRSE = "DeploymentIsNotInReadyStateException";
const _DIRE = "DeploymentIdRequiredException";
const _DLEE = "DeploymentLimitExceededException";
const _DNSE = "DeploymentNotStartedException";
const _DO = "DeploymentOverview";
const _DOPI = "DeregisterOnPremisesInstance";
const _DOPII = "DeregisterOnPremisesInstanceInput";
const _DRBEI = "DeleteResourcesByExternalId";
const _DRBEII = "DeleteResourcesByExternalIdInput";
const _DRBEIO = "DeleteResourcesByExternalIdOutput";
const _DRO = "DeploymentReadyOption";
const _DS = "DeploymentStyle";
const _DT = "DeploymentTarget";
const _DTDNEE = "DeploymentTargetDoesNotExistException";
const _DTIRE = "DeploymentTargetIdRequiredException";
const _DTL = "DeploymentTargetList";
const _DTLE = "DescriptionTooLongException";
const _DTLSEE = "DeploymentTargetListSizeExceededException";
const _ECSS = "ECSService";
const _ECSSL = "ECSServiceList";
const _ECSSMLEE = "ECSServiceMappingLimitExceededException";
const _ECST = "ECSTarget";
const _ECSTS = "ECSTaskSet";
const _ECSTSL = "ECSTaskSetList";
const _ECTF = "EC2TagFilter";
const _ECTFL = "EC2TagFilterList";
const _ECTS = "EC2TagSet";
const _ECTSL = "EC2TagSetList";
const _EI = "ErrorInformation";
const _ELBI = "ELBInfo";
const _ELBIL = "ELBInfoList";
const _F = "Failed";
const _GA = "GetApplication";
const _GAI = "GetApplicationInput";
const _GAO = "GetApplicationOutput";
const _GAR = "GetApplicationRevision";
const _GARI = "GetApplicationRevisionInput";
const _GARO = "GetApplicationRevisionOutput";
const _GD = "GetDeployment";
const _GDC = "GetDeploymentConfig";
const _GDCI = "GetDeploymentConfigInput";
const _GDCO = "GetDeploymentConfigOutput";
const _GDG = "GetDeploymentGroup";
const _GDGI = "GetDeploymentGroupInput";
const _GDGO = "GetDeploymentGroupOutput";
const _GDI = "GetDeploymentInput";
const _GDII = "GetDeploymentInstanceInput";
const _GDIO = "GetDeploymentInstanceOutput";
const _GDIe = "GetDeploymentInstance";
const _GDO = "GetDeploymentOutput";
const _GDT = "GetDeploymentTarget";
const _GDTI = "GetDeploymentTargetInput";
const _GDTO = "GetDeploymentTargetOutput";
const _GFPO = "GreenFleetProvisioningOption";
const _GHATDNEE = "GitHubAccountTokenDoesNotExistException";
const _GHATNRE = "GitHubAccountTokenNameRequiredException";
const _GHL = "GitHubLocation";
const _GOPI = "GetOnPremisesInstance";
const _GOPII = "GetOnPremisesInstanceInput";
const _GOPIO = "GetOnPremisesInstanceOutput";
const _GRI = "GenericRevisionInfo";
const _IACE = "InvalidAlarmConfigException";
const _IAE = "InvalidArnException";
const _IANE = "InvalidApplicationNameException";
const _IARCE = "InvalidAutoRollbackConfigException";
const _IARE = "IamArnRequiredException";
const _IASGE = "InvalidAutoScalingGroupException";
const _IBGDCE = "InvalidBlueGreenDeploymentConfigurationException";
const _IBNFE = "InvalidBucketNameFilterException";
const _ICPE = "InvalidComputePlatformException";
const _IDCNE = "InvalidDeploymentConfigNameException";
const _IDGNE = "InvalidDeploymentGroupNameException";
const _IDIE = "InvalidDeploymentIdException";
const _IDITE = "InvalidDeploymentInstanceTypeException";
const _IDNEE = "InstanceDoesNotExistException";
const _IDSE = "InvalidDeploymentStatusException";
const _IDSEn = "InvalidDeploymentStyleException";
const _IDSFE = "InvalidDeployedStateFilterException";
const _IDTIE = "InvalidDeploymentTargetIdException";
const _IDWTE = "InvalidDeploymentWaitTypeException";
const _IECSSE = "InvalidECSServiceException";
const _IECTCE = "InvalidEC2TagCombinationException";
const _IECTE = "InvalidEC2TagException";
const _IEIE = "InvalidExternalIdException";
const _IFEBE = "InvalidFileExistsBehaviorException";
const _IGHATE = "InvalidGitHubAccountTokenException";
const _IGHATNE = "InvalidGitHubAccountTokenNameException";
const _II = "InstanceInfo";
const _IIASFVE = "InvalidIgnoreApplicationStopFailuresValueException";
const _IIE = "InvalidInputException";
const _IIL = "InstanceInfoList";
const _IINE = "InvalidInstanceNameException";
const _IIRE = "InstanceIdRequiredException";
const _IISAE = "InvalidIamSessionArnException";
const _IISE = "InvalidInstanceStatusException";
const _IITE = "InvalidInstanceTypeException";
const _IIUAE = "InvalidIamUserArnException";
const _IKPFE = "InvalidKeyPrefixFilterException";
const _ILBIE = "InvalidLoadBalancerInfoException";
const _ILEE = "InstanceLimitExceededException";
const _ILEHEIE = "InvalidLifecycleEventHookExecutionIdException";
const _ILEHESE = "InvalidLifecycleEventHookExecutionStatusException";
const _IMHHVE = "InvalidMinimumHealthyHostValueException";
const _INARE = "InstanceNameAlreadyRegisteredException";
const _INRE = "InstanceNameRequiredException";
const _INREn = "InstanceNotRegisteredException";
const _INTE = "InvalidNextTokenException";
const _IOE = "InvalidOperationException";
const _IOPTCE = "InvalidOnPremisesTagCombinationException";
const _IP = "InProgress";
const _IRE = "InvalidRevisionException";
const _IREn = "InvalidRoleException";
const _IRSE = "InvalidRegistrationStatusException";
const _IS = "InstanceSummary";
const _ISAARE = "IamSessionArnAlreadyRegisteredException";
const _ISBE = "InvalidSortByException";
const _ISL = "InstanceSummaryList";
const _ISOE = "InvalidSortOrderException";
const _IT = "InstanceTarget";
const _ITCE = "InvalidTriggerConfigException";
const _ITE = "InvalidTagException";
const _ITFE = "InvalidTagFilterException";
const _ITFNE = "InvalidTargetFilterNameException";
const _ITGPE = "InvalidTargetGroupPairException";
const _ITIE = "InvalidTargetInstancesException";
const _ITRCE = "InvalidTrafficRoutingConfigurationException";
const _ITRE = "InvalidTimeRangeException";
const _ITTAE = "InvalidTagsToAddException";
const _IUAARE = "IamUserArnAlreadyRegisteredException";
const _IUARE = "IamUserArnRequiredException";
const _IUOIOVE = "InvalidUpdateOutdatedInstancesOnlyValueException";
const _IZDCE = "InvalidZonalDeploymentConfigurationException";
const _K = "Key";
const _LA = "ListApplications";
const _LAI = "ListApplicationsInput";
const _LAO = "ListApplicationsOutput";
const _LAR = "ListApplicationRevisions";
const _LARI = "ListApplicationRevisionsInput";
const _LARO = "ListApplicationRevisionsOutput";
const _LBI = "LoadBalancerInfo";
const _LD = "ListDeployments";
const _LDC = "ListDeploymentConfigs";
const _LDCI = "ListDeploymentConfigsInput";
const _LDCO = "ListDeploymentConfigsOutput";
const _LDG = "ListDeploymentGroups";
const _LDGI = "ListDeploymentGroupsInput";
const _LDGO = "ListDeploymentGroupsOutput";
const _LDI = "LastDeploymentInfo";
const _LDII = "ListDeploymentInstancesInput";
const _LDIO = "ListDeploymentInstancesOutput";
const _LDIi = "ListDeploymentsInput";
const _LDIis = "ListDeploymentInstances";
const _LDO = "ListDeploymentsOutput";
const _LDT = "ListDeploymentTargets";
const _LDTI = "ListDeploymentTargetsInput";
const _LDTO = "ListDeploymentTargetsOutput";
const _LE = "LifecycleEvent";
const _LEACE = "LifecycleEventAlreadyCompletedException";
const _LEL = "LifecycleEventList";
const _LFI = "LambdaFunctionInfo";
const _LGHATN = "ListGitHubAccountTokenNames";
const _LGHATNI = "ListGitHubAccountTokenNamesInput";
const _LGHATNO = "ListGitHubAccountTokenNamesOutput";
const _LHLEE = "LifecycleHookLimitExceededException";
const _LOPI = "ListOnPremisesInstances";
const _LOPII = "ListOnPremisesInstancesInput";
const _LOPIO = "ListOnPremisesInstancesOutput";
const _LT = "LambdaTarget";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _MHH = "MinimumHealthyHosts";
const _MHHPZ = "MinimumHealthyHostsPerZone";
const _MIAPE = "MultipleIamArnsProvidedException";
const _NT = "NextToken";
const _ONSE = "OperationNotSupportedException";
const _OPTS = "OnPremisesTagSet";
const _OPTSL = "OnPremisesTagSetList";
const _P = "Pending";
const _PLEHES = "PutLifecycleEventHookExecutionStatus";
const _PLEHESI = "PutLifecycleEventHookExecutionStatusInput";
const _PLEHESO = "PutLifecycleEventHookExecutionStatusOutput";
const _R = "Ready";
const _RA = "ResourceArn";
const _RAR = "RegisterApplicationRevision";
const _RARE = "ResourceArnRequiredException";
const _RARI = "RegisterApplicationRevisionInput";
const _RD = "RelatedDeployments";
const _RDNEE = "RevisionDoesNotExistException";
const _RI = "RevisionInfo";
const _RIL = "RevisionInfoList";
const _RIo = "RollbackInfo";
const _RL = "RevisionLocation";
const _RLL = "RevisionLocationList";
const _ROPI = "RegisterOnPremisesInstance";
const _ROPII = "RegisterOnPremisesInstanceInput";
const _RRE = "RevisionRequiredException";
const _RREo = "RoleRequiredException";
const _RS = "RawString";
const _RTFOPI = "RemoveTagsFromOnPremisesInstances";
const _RTFOPII = "RemoveTagsFromOnPremisesInstancesInput";
const _RVE = "ResourceValidationException";
const _S = "Succeeded";
const _SD = "StopDeployment";
const _SDI = "StopDeploymentInput";
const _SDO = "StopDeploymentOutput";
const _SL = "S3Location";
const _SWTFIT = "SkipWaitTimeForInstanceTermination";
const _SWTFITI = "SkipWaitTimeForInstanceTerminationInput";
const _Sk = "Skipped";
const _T = "Type";
const _TBC = "TimeBasedCanary";
const _TBL = "TimeBasedLinear";
const _TC = "TriggerConfig";
const _TCL = "TriggerConfigList";
const _TE = "ThrottlingException";
const _TF = "TargetFilters";
const _TFL = "TagFilterList";
const _TFa = "TagFilter";
const _TGI = "TargetGroupInfo";
const _TGIL = "TargetGroupInfoList";
const _TGPI = "TargetGroupPairInfo";
const _TGPIL = "TargetGroupPairInfoList";
const _TI = "TargetInstances";
const _TK = "TagKeys";
const _TL = "TagList";
const _TLEE = "TagLimitExceededException";
const _TR = "TimeRange";
const _TRC = "TrafficRoutingConfig";
const _TRE = "TagRequiredException";
const _TRI = "TagResourceInput";
const _TRO = "TagResourceOutput";
const _TRa = "TagResource";
const _TRr = "TrafficRoute";
const _TSLLEE = "TagSetListLimitExceededException";
const _TTLEE = "TriggerTargetsLimitExceededException";
const _Ta = "Tags";
const _Tag = "Tag";
const _UA = "UpdateApplication";
const _UAFDTE = "UnsupportedActionForDeploymentTypeException";
const _UAI = "UpdateApplicationInput";
const _UDG = "UpdateDeploymentGroup";
const _UDGI = "UpdateDeploymentGroupInput";
const _UDGO = "UpdateDeploymentGroupOutput";
const _UR = "UntagResource";
const _URI = "UntagResourceInput";
const _URO = "UntagResourceOutput";
const _V = "Value";
const _ZC = "ZonalConfig";
const _a = "alarms";
const _aC = "alarmConfiguration";
const _aDSI = "additionalDeploymentStatusInfo";
const _aI = "applicationId";
const _aIp = "applicationsInfo";
const _aN = "applicationName";
const _aNp = "applicationNames";
const _aOT = "actionOnTimeout";
const _aRC = "autoRollbackConfiguration";
const _aRE = "autoRollbackEnabled";
const _aSC = "appSpecContent";
const _aSG = "autoScalingGroups";
const _aUOIDI = "autoUpdateOutdatedInstancesDeploymentIds";
const _aUOIRDI = "autoUpdateOutdatedInstancesRootDeploymentId";
const _ac = "action";
const _ap = "application";
const _app = "applications";
const _b = "bucket";
const _bGDC = "blueGreenDeploymentConfiguration";
const _bT = "bundleType";
const _c = "client";
const _cDGN = "currentDeploymentGroupName";
const _cFT = "cloudFormationTarget";
const _cI = "commitId";
const _cIa = "canaryInterval";
const _cN = "clusterName";
const _cP = "computePlatform";
const _cPa = "canaryPercentage";
const _cT = "createTime";
const _cTR = "createTimeRange";
const _cTo = "completeTime";
const _cV = "currentVersion";
const _co = "content";
const _cod = "code";
const _cr = "creator";
const _d = "description";
const _dC = "desiredCount";
const _dCI = "deploymentConfigId";
const _dCIe = "deploymentConfigInfo";
const _dCL = "deploymentConfigsList";
const _dCN = "deploymentConfigName";
const _dG = "deploymentGroups";
const _dGI = "deploymentGroupsInfo";
const _dGIe = "deploymentGroupId";
const _dGIep = "deploymentGroupInfo";
const _dGN = "deploymentGroupNames";
const _dGNe = "deploymentGroupName";
const _dI = "deploymentId";
const _dIe = "deploymentIds";
const _dIep = "deploymentsInfo";
const _dIepl = "deploymentInfo";
const _dO = "deploymentOverview";
const _dOe = "deploymentOption";
const _dRO = "deploymentReadyOption";
const _dS = "deploymentStyle";
const _dSM = "deploymentStatusMessages";
const _dT = "deploymentTargets";
const _dTT = "deploymentTargetType";
const _dTe = "deploymentType";
const _dTep = "deploymentTarget";
const _dTer = "deregisterTime";
const _dWT = "deploymentWaitType";
const _de = "deployed";
const _dep = "deployments";
const _di = "diagnostics";
const _e = "enabled";
const _eC = "errorCode";
const _eI = "externalId";
const _eIL = "elbInfoList";
const _eIr = "errorInformation";
const _eM = "errorMessage";
const _eS = "ecsServices";
const _eT = "ecsTarget";
const _eTF = "ec2TagFilters";
const _eTS = "ec2TagSet";
const _eTSL = "ec2TagSetList";
const _eTa = "eTag";
const _eTn = "endTime";
const _en = "end";
const _er = "error";
const _ev = "events";
const _fA = "functionAlias";
const _fEB = "fileExistsBehavior";
const _fN = "functionName";
const _fUT = "firstUsedTime";
const _fZMDIS = "firstZoneMonitorDurationInSeconds";
const _gFPO = "greenFleetProvisioningOption";
const _gHAN = "gitHubAccountName";
const _gHL = "gitHubLocation";
const _gRI = "genericRevisionInfo";
const _h = "hook";
const _hNCU = "hooksNotCleanedUp";
const _i = "identifer";
const _iA = "instanceArn";
const _iASF = "ignoreApplicationStopFailures";
const _iI = "instanceIds";
const _iIn = "instanceInfos";
const _iIns = "instanceId";
const _iInst = "instanceInfo";
const _iL = "instanceLabel";
const _iLn = "instancesList";
const _iN = "instanceNames";
const _iNn = "instanceName";
const _iOS = "includeOnlyStatuses";
const _iPAF = "ignorePollAlarmFailure";
const _iS = "instancesSummary";
const _iSA = "iamSessionArn";
const _iSF = "instanceStatusFilter";
const _iSn = "instanceSummary";
const _iT = "instanceTarget";
const _iTF = "instanceTypeFilter";
const _iTWTS = "instanceTerminationWaitTimeStarted";
const _iTn = "instanceType";
const _iUA = "iamUserArn";
const _k = "key";
const _lA = "listenerArns";
const _lAD = "lastAttemptedDeployment";
const _lBI = "loadBalancerInfo";
const _lE = "lifecycleEvents";
const _lEHEI = "lifecycleEventHookExecutionId";
const _lEN = "lifecycleEventName";
const _lFI = "lambdaFunctionInfo";
const _lI = "linearInterval";
const _lP = "linearPercentage";
const _lSD = "lastSuccessfulDeployment";
const _lT = "lambdaTarget";
const _lTGH = "linkedToGitHub";
const _lTo = "logTail";
const _lUA = "lastUpdatedAt";
const _lUT = "lastUsedTime";
const _m = "message";
const _mDIS = "monitorDurationInSeconds";
const _mHH = "minimumHealthyHosts";
const _mHHPZ = "minimumHealthyHostsPerZone";
const _n = "name";
const _nAN = "newApplicationName";
const _nDGN = "newDeploymentGroupName";
const _nT = "nextToken";
const _oAC = "overrideAlarmConfiguration";
const _oIS = "outdatedInstancesStrategy";
const _oPITF = "onPremisesInstanceTagFilters";
const _oPTS = "onPremisesTagSet";
const _oPTSL = "onPremisesTagSetList";
const _pC = "pendingCount";
const _pR = "previousRevision";
const _pTR = "prodTrafficRoute";
const _r = "revisions";
const _rC = "runningCount";
const _rD = "relatedDeployments";
const _rDI = "rollbackDeploymentId";
const _rI = "rollbackInfo";
const _rIe = "revisionInfo";
const _rL = "revisionLocation";
const _rM = "rollbackMessage";
const _rS = "registrationStatus";
const _rT = "resourceType";
const _rTDI = "rollbackTriggeringDeploymentId";
const _rTe = "registerTime";
const _rTev = "revisionType";
const _re = "revision";
const _rep = "repository";
const _s = "sha256";
const _sB = "sortBy";
const _sBu = "s3Bucket";
const _sKP = "s3KeyPrefix";
const _sL = "s3Location";
const _sM = "statusMessage";
const _sN = "scriptName";
const _sNe = "serviceName";
const _sO = "sortOrder";
const _sRA = "serviceRoleArn";
const _sT = "startTime";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.codedeploy";
const _st = "status";
const _sta = "start";
const _str = "string";
const _t = "tags";
const _tA = "targetArn";
const _tBC = "timeBasedCanary";
const _tBIODS = "terminateBlueInstancesOnDeploymentSuccess";
const _tBL = "timeBasedLinear";
const _tC = "triggerConfigurations";
const _tE = "triggerEvents";
const _tF = "targetFilters";
const _tFa = "tagFilters";
const _tG = "targetGroup";
const _tGIL = "targetGroupInfoList";
const _tGPIL = "targetGroupPairInfoList";
const _tGa = "targetGroups";
const _tH = "terminationHook";
const _tHE = "terminationHookEnabled";
const _tI = "targetIds";
const _tIa = "targetId";
const _tIar = "targetInstances";
const _tN = "tokenName";
const _tNL = "tokenNameList";
const _tNr = "triggerName";
const _tR = "targetRevision";
const _tRC = "trafficRoutingConfig";
const _tSI = "taskSetsInfo";
const _tSL = "taskSetLabel";
const _tTA = "triggerTargetArn";
const _tTR = "testTrafficRoute";
const _tV = "targetVersion";
const _tVW = "targetVersionWeight";
const _tW = "trafficWeight";
const _tWTIM = "terminationWaitTimeInMinutes";
const _ty = "type";
const _uOIO = "updateOutdatedInstancesOnly";
const _v = "value";
const _ve = "version";
const _wTIM = "waitTimeInMinutes";
const _zC = "zonalConfig";
const n0 = "com.amazonaws.codedeploy";
var AddTagsToOnPremisesInstancesInput = [
    3,
    n0,
    _ATTOPII,
    0,
    [_t, _iN],
    [() => TagList, 64 | 0],
];
var Alarm = [3, n0, _A, 0, [_n], [0]];
var AlarmConfiguration = [3, n0, _AC, 0, [_e, _iPAF, _a], [2, 2, () => AlarmList]];
var AlarmsLimitExceededException = [
    -3,
    n0,
    _ALEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(AlarmsLimitExceededException, AlarmsLimitExceededException$1);
var ApplicationAlreadyExistsException = [
    -3,
    n0,
    _AAEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ApplicationAlreadyExistsException, ApplicationAlreadyExistsException$1);
var ApplicationDoesNotExistException = [
    -3,
    n0,
    _ADNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ApplicationDoesNotExistException, ApplicationDoesNotExistException$1);
var ApplicationInfo = [
    3,
    n0,
    _AI,
    0,
    [_aI, _aN, _cT, _lTGH, _gHAN, _cP],
    [0, 0, 4, 2, 0, 0],
];
var ApplicationLimitExceededException = [
    -3,
    n0,
    _ALEEp,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ApplicationLimitExceededException, ApplicationLimitExceededException$1);
var ApplicationNameRequiredException = [
    -3,
    n0,
    _ANRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ApplicationNameRequiredException, ApplicationNameRequiredException$1);
var AppSpecContent = [3, n0, _ASC, 0, [_co, _s], [0, 0]];
var ArnNotSupportedException = [
    -3,
    n0,
    _ANSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ArnNotSupportedException, ArnNotSupportedException$1);
var AutoRollbackConfiguration = [3, n0, _ARC, 0, [_e, _ev], [2, 64 | 0]];
var AutoScalingGroup = [3, n0, _ASG, 0, [_n, _h, _tH], [0, 0, 0]];
var BatchGetApplicationRevisionsInput = [
    3,
    n0,
    _BGARI,
    0,
    [_aN, _r],
    [0, () => RevisionLocationList],
];
var BatchGetApplicationRevisionsOutput = [
    3,
    n0,
    _BGARO,
    0,
    [_aN, _eM, _r],
    [0, 0, () => RevisionInfoList],
];
var BatchGetApplicationsInput = [3, n0, _BGAI, 0, [_aNp], [64 | 0]];
var BatchGetApplicationsOutput = [3, n0, _BGAO, 0, [_aIp], [() => ApplicationsInfoList]];
var BatchGetDeploymentGroupsInput = [3, n0, _BGDGI, 0, [_aN, _dGN], [0, 64 | 0]];
var BatchGetDeploymentGroupsOutput = [
    3,
    n0,
    _BGDGO,
    0,
    [_dGI, _eM],
    [() => DeploymentGroupInfoList, 0],
];
var BatchGetDeploymentInstancesInput = [3, n0, _BGDII, 0, [_dI, _iI], [0, 64 | 0]];
var BatchGetDeploymentInstancesOutput = [
    3,
    n0,
    _BGDIO,
    0,
    [_iS, _eM],
    [() => InstanceSummaryList, 0],
];
var BatchGetDeploymentsInput = [3, n0, _BGDI, 0, [_dIe], [64 | 0]];
var BatchGetDeploymentsOutput = [3, n0, _BGDO, 0, [_dIep], [() => DeploymentsInfoList]];
var BatchGetDeploymentTargetsInput = [3, n0, _BGDTI, 0, [_dI, _tI], [0, 64 | 0]];
var BatchGetDeploymentTargetsOutput = [
    3,
    n0,
    _BGDTO,
    0,
    [_dT],
    [() => DeploymentTargetList],
];
var BatchGetOnPremisesInstancesInput = [3, n0, _BGOPII, 0, [_iN], [64 | 0]];
var BatchGetOnPremisesInstancesOutput = [
    3,
    n0,
    _BGOPIO,
    0,
    [_iIn],
    [() => InstanceInfoList],
];
var BatchLimitExceededException = [
    -3,
    n0,
    _BLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(BatchLimitExceededException, BatchLimitExceededException$1);
var BlueGreenDeploymentConfiguration = [
    3,
    n0,
    _BGDC,
    0,
    [_tBIODS, _dRO, _gFPO],
    [() => BlueInstanceTerminationOption, () => DeploymentReadyOption, () => GreenFleetProvisioningOption],
];
var BlueInstanceTerminationOption = [3, n0, _BITO, 0, [_ac, _tWTIM], [0, 1]];
var BucketNameFilterRequiredException = [
    -3,
    n0,
    _BNFRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(BucketNameFilterRequiredException, BucketNameFilterRequiredException$1);
var CloudFormationTarget = [
    3,
    n0,
    _CFT,
    0,
    [_dI, _tIa, _lUA, _lE, _st, _rT, _tVW],
    [0, 0, 4, () => LifecycleEventList, 0, 0, 1],
];
var ContinueDeploymentInput = [3, n0, _CDI, 0, [_dI, _dWT], [0, 0]];
var CreateApplicationInput = [3, n0, _CAI, 0, [_aN, _cP, _t], [0, 0, () => TagList]];
var CreateApplicationOutput = [3, n0, _CAO, 0, [_aI], [0]];
var CreateDeploymentConfigInput = [
    3,
    n0,
    _CDCI,
    0,
    [_dCN, _mHH, _tRC, _cP, _zC],
    [0, () => MinimumHealthyHosts, () => TrafficRoutingConfig, 0, () => ZonalConfig],
];
var CreateDeploymentConfigOutput = [3, n0, _CDCO, 0, [_dCI], [0]];
var CreateDeploymentGroupInput = [
    3,
    n0,
    _CDGI,
    0,
    [_aN, _dGNe, _dCN, _eTF, _oPITF, _aSG, _sRA, _tC, _aC, _aRC, _oIS, _dS, _bGDC, _lBI, _eTS, _eS, _oPTS, _t, _tHE],
    [
        0,
        0,
        0,
        () => EC2TagFilterList,
        () => TagFilterList,
        64 | 0,
        0,
        () => TriggerConfigList,
        () => AlarmConfiguration,
        () => AutoRollbackConfiguration,
        0,
        () => DeploymentStyle,
        () => BlueGreenDeploymentConfiguration,
        () => LoadBalancerInfo,
        () => EC2TagSet,
        () => ECSServiceList,
        () => OnPremisesTagSet,
        () => TagList,
        2,
    ],
];
var CreateDeploymentGroupOutput = [3, n0, _CDGO, 0, [_dGIe], [0]];
var CreateDeploymentInput = [
    3,
    n0,
    _CDIr,
    0,
    [_aN, _dGNe, _re, _dCN, _d, _iASF, _tIar, _aRC, _uOIO, _fEB, _oAC],
    [
        0,
        0,
        () => RevisionLocation,
        0,
        0,
        2,
        () => TargetInstances,
        () => AutoRollbackConfiguration,
        2,
        0,
        () => AlarmConfiguration,
    ],
];
var CreateDeploymentOutput = [3, n0, _CDO, 0, [_dI], [0]];
var DeleteApplicationInput = [3, n0, _DAI, 0, [_aN], [0]];
var DeleteDeploymentConfigInput = [3, n0, _DDCI, 0, [_dCN], [0]];
var DeleteDeploymentGroupInput = [3, n0, _DDGI, 0, [_aN, _dGNe], [0, 0]];
var DeleteDeploymentGroupOutput = [
    3,
    n0,
    _DDGO,
    0,
    [_hNCU],
    [() => AutoScalingGroupList],
];
var DeleteGitHubAccountTokenInput = [3, n0, _DGHATI, 0, [_tN], [0]];
var DeleteGitHubAccountTokenOutput = [3, n0, _DGHATO, 0, [_tN], [0]];
var DeleteResourcesByExternalIdInput = [3, n0, _DRBEII, 0, [_eI], [0]];
var DeleteResourcesByExternalIdOutput = [3, n0, _DRBEIO, 0, [], []];
var DeploymentAlreadyCompletedException = [
    -3,
    n0,
    _DACE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentAlreadyCompletedException, DeploymentAlreadyCompletedException$1);
var DeploymentConfigAlreadyExistsException = [
    -3,
    n0,
    _DCAEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentConfigAlreadyExistsException, DeploymentConfigAlreadyExistsException$1);
var DeploymentConfigDoesNotExistException = [
    -3,
    n0,
    _DCDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentConfigDoesNotExistException, DeploymentConfigDoesNotExistException$1);
var DeploymentConfigInfo = [
    3,
    n0,
    _DCI,
    0,
    [_dCI, _dCN, _mHH, _cT, _cP, _tRC, _zC],
    [0, 0, () => MinimumHealthyHosts, 4, 0, () => TrafficRoutingConfig, () => ZonalConfig],
];
var DeploymentConfigInUseException = [
    -3,
    n0,
    _DCIUE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentConfigInUseException, DeploymentConfigInUseException$1);
var DeploymentConfigLimitExceededException = [
    -3,
    n0,
    _DCLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentConfigLimitExceededException, DeploymentConfigLimitExceededException$1);
var DeploymentConfigNameRequiredException = [
    -3,
    n0,
    _DCNRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentConfigNameRequiredException, DeploymentConfigNameRequiredException$1);
var DeploymentDoesNotExistException = [
    -3,
    n0,
    _DDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentDoesNotExistException, DeploymentDoesNotExistException$1);
var DeploymentGroupAlreadyExistsException = [
    -3,
    n0,
    _DGAEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentGroupAlreadyExistsException, DeploymentGroupAlreadyExistsException$1);
var DeploymentGroupDoesNotExistException = [
    -3,
    n0,
    _DGDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentGroupDoesNotExistException, DeploymentGroupDoesNotExistException$1);
var DeploymentGroupInfo = [
    3,
    n0,
    _DGI,
    0,
    [
        _aN,
        _dGIe,
        _dGNe,
        _dCN,
        _eTF,
        _oPITF,
        _aSG,
        _sRA,
        _tR,
        _tC,
        _aC,
        _aRC,
        _dS,
        _oIS,
        _bGDC,
        _lBI,
        _lSD,
        _lAD,
        _eTS,
        _oPTS,
        _cP,
        _eS,
        _tHE,
    ],
    [
        0,
        0,
        0,
        0,
        () => EC2TagFilterList,
        () => TagFilterList,
        () => AutoScalingGroupList,
        0,
        () => RevisionLocation,
        () => TriggerConfigList,
        () => AlarmConfiguration,
        () => AutoRollbackConfiguration,
        () => DeploymentStyle,
        0,
        () => BlueGreenDeploymentConfiguration,
        () => LoadBalancerInfo,
        () => LastDeploymentInfo,
        () => LastDeploymentInfo,
        () => EC2TagSet,
        () => OnPremisesTagSet,
        0,
        () => ECSServiceList,
        2,
    ],
];
var DeploymentGroupLimitExceededException = [
    -3,
    n0,
    _DGLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentGroupLimitExceededException, DeploymentGroupLimitExceededException$1);
var DeploymentGroupNameRequiredException = [
    -3,
    n0,
    _DGNRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentGroupNameRequiredException, DeploymentGroupNameRequiredException$1);
var DeploymentIdRequiredException = [
    -3,
    n0,
    _DIRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentIdRequiredException, DeploymentIdRequiredException$1);
var DeploymentInfo = [
    3,
    n0,
    _DI,
    0,
    [
        _aN,
        _dGNe,
        _dCN,
        _dI,
        _pR,
        _re,
        _st,
        _eIr,
        _cT,
        _sT,
        _cTo,
        _dO,
        _d,
        _cr,
        _iASF,
        _aRC,
        _uOIO,
        _rI,
        _dS,
        _tIar,
        _iTWTS,
        _bGDC,
        _lBI,
        _aDSI,
        _fEB,
        _dSM,
        _cP,
        _eI,
        _rD,
        _oAC,
    ],
    [
        0,
        0,
        0,
        0,
        () => RevisionLocation,
        () => RevisionLocation,
        0,
        () => ErrorInformation,
        4,
        4,
        4,
        () => DeploymentOverview,
        0,
        0,
        2,
        () => AutoRollbackConfiguration,
        2,
        () => RollbackInfo,
        () => DeploymentStyle,
        () => TargetInstances,
        2,
        () => BlueGreenDeploymentConfiguration,
        () => LoadBalancerInfo,
        0,
        0,
        64 | 0,
        0,
        0,
        () => RelatedDeployments,
        () => AlarmConfiguration,
    ],
];
var DeploymentIsNotInReadyStateException = [
    -3,
    n0,
    _DINIRSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentIsNotInReadyStateException, DeploymentIsNotInReadyStateException$1);
var DeploymentLimitExceededException = [
    -3,
    n0,
    _DLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentLimitExceededException, DeploymentLimitExceededException$1);
var DeploymentNotStartedException = [
    -3,
    n0,
    _DNSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentNotStartedException, DeploymentNotStartedException$1);
var DeploymentOverview = [3, n0, _DO, 0, [_P, _IP, _S, _F, _Sk, _R], [1, 1, 1, 1, 1, 1]];
var DeploymentReadyOption = [3, n0, _DRO, 0, [_aOT, _wTIM], [0, 1]];
var DeploymentStyle = [3, n0, _DS, 0, [_dTe, _dOe], [0, 0]];
var DeploymentTarget = [
    3,
    n0,
    _DT,
    0,
    [_dTT, _iT, _lT, _eT, _cFT],
    [0, () => InstanceTarget, () => LambdaTarget, () => ECSTarget, () => CloudFormationTarget],
];
var DeploymentTargetDoesNotExistException = [
    -3,
    n0,
    _DTDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentTargetDoesNotExistException, DeploymentTargetDoesNotExistException$1);
var DeploymentTargetIdRequiredException = [
    -3,
    n0,
    _DTIRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentTargetIdRequiredException, DeploymentTargetIdRequiredException$1);
var DeploymentTargetListSizeExceededException = [
    -3,
    n0,
    _DTLSEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeploymentTargetListSizeExceededException, DeploymentTargetListSizeExceededException$1);
var DeregisterOnPremisesInstanceInput = [3, n0, _DOPII, 0, [_iNn], [0]];
var DescriptionTooLongException = [
    -3,
    n0,
    _DTLE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DescriptionTooLongException, DescriptionTooLongException$1);
var Diagnostics = [3, n0, _D, 0, [_eC, _sN, _m, _lTo], [0, 0, 0, 0]];
var EC2TagFilter = [3, n0, _ECTF, 0, [_K, _V, _T], [0, 0, 0]];
var EC2TagSet = [3, n0, _ECTS, 0, [_eTSL], [() => EC2TagSetList]];
var ECSService = [3, n0, _ECSS, 0, [_sNe, _cN], [0, 0]];
var ECSServiceMappingLimitExceededException = [
    -3,
    n0,
    _ECSSMLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ECSServiceMappingLimitExceededException, ECSServiceMappingLimitExceededException$1);
var ECSTarget = [
    3,
    n0,
    _ECST,
    0,
    [_dI, _tIa, _tA, _lUA, _lE, _st, _tSI],
    [0, 0, 0, 4, () => LifecycleEventList, 0, () => ECSTaskSetList],
];
var ECSTaskSet = [
    3,
    n0,
    _ECSTS,
    0,
    [_i, _dC, _pC, _rC, _st, _tW, _tG, _tSL],
    [0, 1, 1, 1, 0, 1, () => TargetGroupInfo, 0],
];
var ELBInfo = [3, n0, _ELBI, 0, [_n], [0]];
var ErrorInformation = [3, n0, _EI, 0, [_cod, _m], [0, 0]];
var GenericRevisionInfo = [
    3,
    n0,
    _GRI,
    0,
    [_d, _dG, _fUT, _lUT, _rTe],
    [0, 64 | 0, 4, 4, 4],
];
var GetApplicationInput = [3, n0, _GAI, 0, [_aN], [0]];
var GetApplicationOutput = [3, n0, _GAO, 0, [_ap], [() => ApplicationInfo]];
var GetApplicationRevisionInput = [
    3,
    n0,
    _GARI,
    0,
    [_aN, _re],
    [0, () => RevisionLocation],
];
var GetApplicationRevisionOutput = [
    3,
    n0,
    _GARO,
    0,
    [_aN, _re, _rIe],
    [0, () => RevisionLocation, () => GenericRevisionInfo],
];
var GetDeploymentConfigInput = [3, n0, _GDCI, 0, [_dCN], [0]];
var GetDeploymentConfigOutput = [3, n0, _GDCO, 0, [_dCIe], [() => DeploymentConfigInfo]];
var GetDeploymentGroupInput = [3, n0, _GDGI, 0, [_aN, _dGNe], [0, 0]];
var GetDeploymentGroupOutput = [3, n0, _GDGO, 0, [_dGIep], [() => DeploymentGroupInfo]];
var GetDeploymentInput = [3, n0, _GDI, 0, [_dI], [0]];
var GetDeploymentInstanceInput = [3, n0, _GDII, 0, [_dI, _iIns], [0, 0]];
var GetDeploymentInstanceOutput = [3, n0, _GDIO, 0, [_iSn], [() => InstanceSummary]];
var GetDeploymentOutput = [3, n0, _GDO, 0, [_dIepl], [() => DeploymentInfo]];
var GetDeploymentTargetInput = [3, n0, _GDTI, 0, [_dI, _tIa], [0, 0]];
var GetDeploymentTargetOutput = [3, n0, _GDTO, 0, [_dTep], [() => DeploymentTarget]];
var GetOnPremisesInstanceInput = [3, n0, _GOPII, 0, [_iNn], [0]];
var GetOnPremisesInstanceOutput = [3, n0, _GOPIO, 0, [_iInst], [() => InstanceInfo]];
var GitHubAccountTokenDoesNotExistException = [
    -3,
    n0,
    _GHATDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(GitHubAccountTokenDoesNotExistException, GitHubAccountTokenDoesNotExistException$1);
var GitHubAccountTokenNameRequiredException = [
    -3,
    n0,
    _GHATNRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(GitHubAccountTokenNameRequiredException, GitHubAccountTokenNameRequiredException$1);
var GitHubLocation = [3, n0, _GHL, 0, [_rep, _cI], [0, 0]];
var GreenFleetProvisioningOption = [3, n0, _GFPO, 0, [_ac], [0]];
var IamArnRequiredException = [
    -3,
    n0,
    _IARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IamArnRequiredException, IamArnRequiredException$1);
var IamSessionArnAlreadyRegisteredException = [
    -3,
    n0,
    _ISAARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IamSessionArnAlreadyRegisteredException, IamSessionArnAlreadyRegisteredException$1);
var IamUserArnAlreadyRegisteredException = [
    -3,
    n0,
    _IUAARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IamUserArnAlreadyRegisteredException, IamUserArnAlreadyRegisteredException$1);
var IamUserArnRequiredException = [
    -3,
    n0,
    _IUARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IamUserArnRequiredException, IamUserArnRequiredException$1);
var InstanceDoesNotExistException = [
    -3,
    n0,
    _IDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceDoesNotExistException, InstanceDoesNotExistException$1);
var InstanceIdRequiredException = [
    -3,
    n0,
    _IIRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceIdRequiredException, InstanceIdRequiredException$1);
var InstanceInfo = [
    3,
    n0,
    _II,
    0,
    [_iNn, _iSA, _iUA, _iA, _rTe, _dTer, _t],
    [0, 0, 0, 0, 4, 4, () => TagList],
];
var InstanceLimitExceededException = [
    -3,
    n0,
    _ILEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceLimitExceededException, InstanceLimitExceededException$1);
var InstanceNameAlreadyRegisteredException = [
    -3,
    n0,
    _INARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceNameAlreadyRegisteredException, InstanceNameAlreadyRegisteredException$1);
var InstanceNameRequiredException = [
    -3,
    n0,
    _INRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceNameRequiredException, InstanceNameRequiredException$1);
var InstanceNotRegisteredException = [
    -3,
    n0,
    _INREn,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InstanceNotRegisteredException, InstanceNotRegisteredException$1);
var InstanceSummary = [
    3,
    n0,
    _IS,
    0,
    [_dI, _iIns, _st, _lUA, _lE, _iTn],
    [0, 0, 0, 4, () => LifecycleEventList, 0],
];
var InstanceTarget = [
    3,
    n0,
    _IT,
    0,
    [_dI, _tIa, _tA, _st, _lUA, _lE, _iL],
    [0, 0, 0, 0, 4, () => LifecycleEventList, 0],
];
var InvalidAlarmConfigException = [
    -3,
    n0,
    _IACE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidAlarmConfigException, InvalidAlarmConfigException$1);
var InvalidApplicationNameException = [
    -3,
    n0,
    _IANE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidApplicationNameException, InvalidApplicationNameException$1);
var InvalidArnException = [
    -3,
    n0,
    _IAE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidArnException, InvalidArnException$1);
var InvalidAutoRollbackConfigException = [
    -3,
    n0,
    _IARCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidAutoRollbackConfigException, InvalidAutoRollbackConfigException$1);
var InvalidAutoScalingGroupException = [
    -3,
    n0,
    _IASGE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidAutoScalingGroupException, InvalidAutoScalingGroupException$1);
var InvalidBlueGreenDeploymentConfigurationException = [
    -3,
    n0,
    _IBGDCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidBlueGreenDeploymentConfigurationException, InvalidBlueGreenDeploymentConfigurationException$1);
var InvalidBucketNameFilterException = [
    -3,
    n0,
    _IBNFE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidBucketNameFilterException, InvalidBucketNameFilterException$1);
var InvalidComputePlatformException = [
    -3,
    n0,
    _ICPE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidComputePlatformException, InvalidComputePlatformException$1);
var InvalidDeployedStateFilterException = [
    -3,
    n0,
    _IDSFE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeployedStateFilterException, InvalidDeployedStateFilterException$1);
var InvalidDeploymentConfigNameException = [
    -3,
    n0,
    _IDCNE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentConfigNameException, InvalidDeploymentConfigNameException$1);
var InvalidDeploymentGroupNameException = [
    -3,
    n0,
    _IDGNE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentGroupNameException, InvalidDeploymentGroupNameException$1);
var InvalidDeploymentIdException = [
    -3,
    n0,
    _IDIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentIdException, InvalidDeploymentIdException$1);
var InvalidDeploymentInstanceTypeException = [
    -3,
    n0,
    _IDITE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentInstanceTypeException, InvalidDeploymentInstanceTypeException$1);
var InvalidDeploymentStatusException = [
    -3,
    n0,
    _IDSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentStatusException, InvalidDeploymentStatusException$1);
var InvalidDeploymentStyleException = [
    -3,
    n0,
    _IDSEn,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentStyleException, InvalidDeploymentStyleException$1);
var InvalidDeploymentTargetIdException = [
    -3,
    n0,
    _IDTIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentTargetIdException, InvalidDeploymentTargetIdException$1);
var InvalidDeploymentWaitTypeException = [
    -3,
    n0,
    _IDWTE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDeploymentWaitTypeException, InvalidDeploymentWaitTypeException$1);
var InvalidEC2TagCombinationException = [
    -3,
    n0,
    _IECTCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidEC2TagCombinationException, InvalidEC2TagCombinationException$1);
var InvalidEC2TagException = [
    -3,
    n0,
    _IECTE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidEC2TagException, InvalidEC2TagException$1);
var InvalidECSServiceException = [
    -3,
    n0,
    _IECSSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidECSServiceException, InvalidECSServiceException$1);
var InvalidExternalIdException = [
    -3,
    n0,
    _IEIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidExternalIdException, InvalidExternalIdException$1);
var InvalidFileExistsBehaviorException = [
    -3,
    n0,
    _IFEBE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidFileExistsBehaviorException, InvalidFileExistsBehaviorException$1);
var InvalidGitHubAccountTokenException = [
    -3,
    n0,
    _IGHATE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidGitHubAccountTokenException, InvalidGitHubAccountTokenException$1);
var InvalidGitHubAccountTokenNameException = [
    -3,
    n0,
    _IGHATNE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidGitHubAccountTokenNameException, InvalidGitHubAccountTokenNameException$1);
var InvalidIamSessionArnException = [
    -3,
    n0,
    _IISAE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidIamSessionArnException, InvalidIamSessionArnException$1);
var InvalidIamUserArnException = [
    -3,
    n0,
    _IIUAE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidIamUserArnException, InvalidIamUserArnException$1);
var InvalidIgnoreApplicationStopFailuresValueException = [
    -3,
    n0,
    _IIASFVE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidIgnoreApplicationStopFailuresValueException, InvalidIgnoreApplicationStopFailuresValueException$1);
var InvalidInputException = [
    -3,
    n0,
    _IIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidInputException, InvalidInputException$1);
var InvalidInstanceNameException = [
    -3,
    n0,
    _IINE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidInstanceNameException, InvalidInstanceNameException$1);
var InvalidInstanceStatusException = [
    -3,
    n0,
    _IISE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidInstanceStatusException, InvalidInstanceStatusException$1);
var InvalidInstanceTypeException = [
    -3,
    n0,
    _IITE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidInstanceTypeException, InvalidInstanceTypeException$1);
var InvalidKeyPrefixFilterException = [
    -3,
    n0,
    _IKPFE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidKeyPrefixFilterException, InvalidKeyPrefixFilterException$1);
var InvalidLifecycleEventHookExecutionIdException = [
    -3,
    n0,
    _ILEHEIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidLifecycleEventHookExecutionIdException, InvalidLifecycleEventHookExecutionIdException$1);
var InvalidLifecycleEventHookExecutionStatusException = [
    -3,
    n0,
    _ILEHESE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidLifecycleEventHookExecutionStatusException, InvalidLifecycleEventHookExecutionStatusException$1);
var InvalidLoadBalancerInfoException = [
    -3,
    n0,
    _ILBIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidLoadBalancerInfoException, InvalidLoadBalancerInfoException$1);
var InvalidMinimumHealthyHostValueException = [
    -3,
    n0,
    _IMHHVE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidMinimumHealthyHostValueException, InvalidMinimumHealthyHostValueException$1);
var InvalidNextTokenException = [
    -3,
    n0,
    _INTE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidNextTokenException, InvalidNextTokenException$1);
var InvalidOnPremisesTagCombinationException = [
    -3,
    n0,
    _IOPTCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidOnPremisesTagCombinationException, InvalidOnPremisesTagCombinationException$1);
var InvalidOperationException = [
    -3,
    n0,
    _IOE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidOperationException, InvalidOperationException$1);
var InvalidRegistrationStatusException = [
    -3,
    n0,
    _IRSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidRegistrationStatusException, InvalidRegistrationStatusException$1);
var InvalidRevisionException = [
    -3,
    n0,
    _IRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidRevisionException, InvalidRevisionException$1);
var InvalidRoleException = [
    -3,
    n0,
    _IREn,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidRoleException, InvalidRoleException$1);
var InvalidSortByException = [
    -3,
    n0,
    _ISBE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSortByException, InvalidSortByException$1);
var InvalidSortOrderException = [
    -3,
    n0,
    _ISOE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSortOrderException, InvalidSortOrderException$1);
var InvalidTagException = [
    -3,
    n0,
    _ITE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTagException, InvalidTagException$1);
var InvalidTagFilterException = [
    -3,
    n0,
    _ITFE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTagFilterException, InvalidTagFilterException$1);
var InvalidTagsToAddException = [
    -3,
    n0,
    _ITTAE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTagsToAddException, InvalidTagsToAddException$1);
var InvalidTargetFilterNameException = [
    -3,
    n0,
    _ITFNE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTargetFilterNameException, InvalidTargetFilterNameException$1);
var InvalidTargetGroupPairException = [
    -3,
    n0,
    _ITGPE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTargetGroupPairException, InvalidTargetGroupPairException$1);
var InvalidTargetInstancesException = [
    -3,
    n0,
    _ITIE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTargetInstancesException, InvalidTargetInstancesException$1);
var InvalidTimeRangeException = [
    -3,
    n0,
    _ITRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTimeRangeException, InvalidTimeRangeException$1);
var InvalidTrafficRoutingConfigurationException = [
    -3,
    n0,
    _ITRCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTrafficRoutingConfigurationException, InvalidTrafficRoutingConfigurationException$1);
var InvalidTriggerConfigException = [
    -3,
    n0,
    _ITCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTriggerConfigException, InvalidTriggerConfigException$1);
var InvalidUpdateOutdatedInstancesOnlyValueException = [
    -3,
    n0,
    _IUOIOVE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidUpdateOutdatedInstancesOnlyValueException, InvalidUpdateOutdatedInstancesOnlyValueException$1);
var InvalidZonalDeploymentConfigurationException = [
    -3,
    n0,
    _IZDCE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidZonalDeploymentConfigurationException, InvalidZonalDeploymentConfigurationException$1);
var LambdaFunctionInfo = [3, n0, _LFI, 0, [_fN, _fA, _cV, _tV, _tVW], [0, 0, 0, 0, 1]];
var LambdaTarget = [
    3,
    n0,
    _LT,
    0,
    [_dI, _tIa, _tA, _st, _lUA, _lE, _lFI],
    [0, 0, 0, 0, 4, () => LifecycleEventList, () => LambdaFunctionInfo],
];
var LastDeploymentInfo = [3, n0, _LDI, 0, [_dI, _st, _eTn, _cT], [0, 0, 4, 4]];
var LifecycleEvent = [
    3,
    n0,
    _LE,
    0,
    [_lEN, _di, _sT, _eTn, _st],
    [0, () => Diagnostics, 4, 4, 0],
];
var LifecycleEventAlreadyCompletedException = [
    -3,
    n0,
    _LEACE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LifecycleEventAlreadyCompletedException, LifecycleEventAlreadyCompletedException$1);
var LifecycleHookLimitExceededException = [
    -3,
    n0,
    _LHLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LifecycleHookLimitExceededException, LifecycleHookLimitExceededException$1);
var ListApplicationRevisionsInput = [
    3,
    n0,
    _LARI,
    0,
    [_aN, _sB, _sO, _sBu, _sKP, _de, _nT],
    [0, 0, 0, 0, 0, 0, 0],
];
var ListApplicationRevisionsOutput = [
    3,
    n0,
    _LARO,
    0,
    [_r, _nT],
    [() => RevisionLocationList, 0],
];
var ListApplicationsInput = [3, n0, _LAI, 0, [_nT], [0]];
var ListApplicationsOutput = [3, n0, _LAO, 0, [_app, _nT], [64 | 0, 0]];
var ListDeploymentConfigsInput = [3, n0, _LDCI, 0, [_nT], [0]];
var ListDeploymentConfigsOutput = [3, n0, _LDCO, 0, [_dCL, _nT], [64 | 0, 0]];
var ListDeploymentGroupsInput = [3, n0, _LDGI, 0, [_aN, _nT], [0, 0]];
var ListDeploymentGroupsOutput = [3, n0, _LDGO, 0, [_aN, _dG, _nT], [0, 64 | 0, 0]];
var ListDeploymentInstancesInput = [
    3,
    n0,
    _LDII,
    0,
    [_dI, _nT, _iSF, _iTF],
    [0, 0, 64 | 0, 64 | 0],
];
var ListDeploymentInstancesOutput = [3, n0, _LDIO, 0, [_iLn, _nT], [64 | 0, 0]];
var ListDeploymentsInput = [
    3,
    n0,
    _LDIi,
    0,
    [_aN, _dGNe, _eI, _iOS, _cTR, _nT],
    [0, 0, 0, 64 | 0, () => TimeRange, 0],
];
var ListDeploymentsOutput = [3, n0, _LDO, 0, [_dep, _nT], [64 | 0, 0]];
var ListDeploymentTargetsInput = [
    3,
    n0,
    _LDTI,
    0,
    [_dI, _nT, _tF],
    [0, 0, [2, n0, _TF, 0, 0, 64 | 0]],
];
var ListDeploymentTargetsOutput = [3, n0, _LDTO, 0, [_tI, _nT], [64 | 0, 0]];
var ListGitHubAccountTokenNamesInput = [3, n0, _LGHATNI, 0, [_nT], [0]];
var ListGitHubAccountTokenNamesOutput = [3, n0, _LGHATNO, 0, [_tNL, _nT], [64 | 0, 0]];
var ListOnPremisesInstancesInput = [
    3,
    n0,
    _LOPII,
    0,
    [_rS, _tFa, _nT],
    [0, () => TagFilterList, 0],
];
var ListOnPremisesInstancesOutput = [3, n0, _LOPIO, 0, [_iN, _nT], [64 | 0, 0]];
var ListTagsForResourceInput = [3, n0, _LTFRI, 0, [_RA, _NT], [0, 0]];
var ListTagsForResourceOutput = [3, n0, _LTFRO, 0, [_Ta, _NT], [() => TagList, 0]];
var LoadBalancerInfo = [
    3,
    n0,
    _LBI,
    0,
    [_eIL, _tGIL, _tGPIL],
    [() => ELBInfoList, () => TargetGroupInfoList, () => TargetGroupPairInfoList],
];
var MinimumHealthyHosts = [3, n0, _MHH, 0, [_ty, _v], [0, 1]];
var MinimumHealthyHostsPerZone = [3, n0, _MHHPZ, 0, [_ty, _v], [0, 1]];
var MultipleIamArnsProvidedException = [
    -3,
    n0,
    _MIAPE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(MultipleIamArnsProvidedException, MultipleIamArnsProvidedException$1);
var OnPremisesTagSet = [3, n0, _OPTS, 0, [_oPTSL], [() => OnPremisesTagSetList]];
var OperationNotSupportedException = [
    -3,
    n0,
    _ONSE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(OperationNotSupportedException, OperationNotSupportedException$1);
var PutLifecycleEventHookExecutionStatusInput = [
    3,
    n0,
    _PLEHESI,
    0,
    [_dI, _lEHEI, _st],
    [0, 0, 0],
];
var PutLifecycleEventHookExecutionStatusOutput = [3, n0, _PLEHESO, 0, [_lEHEI], [0]];
var RawString = [3, n0, _RS, 0, [_co, _s], [0, 0]];
var RegisterApplicationRevisionInput = [
    3,
    n0,
    _RARI,
    0,
    [_aN, _d, _re],
    [0, 0, () => RevisionLocation],
];
var RegisterOnPremisesInstanceInput = [3, n0, _ROPII, 0, [_iNn, _iSA, _iUA], [0, 0, 0]];
var RelatedDeployments = [3, n0, _RD, 0, [_aUOIRDI, _aUOIDI], [0, 64 | 0]];
var RemoveTagsFromOnPremisesInstancesInput = [
    3,
    n0,
    _RTFOPII,
    0,
    [_t, _iN],
    [() => TagList, 64 | 0],
];
var ResourceArnRequiredException = [
    -3,
    n0,
    _RARE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ResourceArnRequiredException, ResourceArnRequiredException$1);
var ResourceValidationException = [
    -3,
    n0,
    _RVE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ResourceValidationException, ResourceValidationException$1);
var RevisionDoesNotExistException = [
    -3,
    n0,
    _RDNEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RevisionDoesNotExistException, RevisionDoesNotExistException$1);
var RevisionInfo = [
    3,
    n0,
    _RI,
    0,
    [_rL, _gRI],
    [() => RevisionLocation, () => GenericRevisionInfo],
];
var RevisionLocation = [
    3,
    n0,
    _RL,
    0,
    [_rTev, _sL, _gHL, _str, _aSC],
    [0, () => S3Location, () => GitHubLocation, () => RawString, () => AppSpecContent],
];
var RevisionRequiredException = [
    -3,
    n0,
    _RRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RevisionRequiredException, RevisionRequiredException$1);
var RoleRequiredException = [
    -3,
    n0,
    _RREo,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RoleRequiredException, RoleRequiredException$1);
var RollbackInfo = [3, n0, _RIo, 0, [_rDI, _rTDI, _rM], [0, 0, 0]];
var S3Location = [3, n0, _SL, 0, [_b, _k, _bT, _ve, _eTa], [0, 0, 0, 0, 0]];
var SkipWaitTimeForInstanceTerminationInput = [3, n0, _SWTFITI, 0, [_dI], [0]];
var StopDeploymentInput = [3, n0, _SDI, 0, [_dI, _aRE], [0, 2]];
var StopDeploymentOutput = [3, n0, _SDO, 0, [_st, _sM], [0, 0]];
var Tag = [3, n0, _Tag, 0, [_K, _V], [0, 0]];
var TagFilter = [3, n0, _TFa, 0, [_K, _V, _T], [0, 0, 0]];
var TagLimitExceededException = [
    -3,
    n0,
    _TLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TagLimitExceededException, TagLimitExceededException$1);
var TagRequiredException = [
    -3,
    n0,
    _TRE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TagRequiredException, TagRequiredException$1);
var TagResourceInput = [3, n0, _TRI, 0, [_RA, _Ta], [0, () => TagList]];
var TagResourceOutput = [3, n0, _TRO, 0, [], []];
var TagSetListLimitExceededException = [
    -3,
    n0,
    _TSLLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TagSetListLimitExceededException, TagSetListLimitExceededException$1);
var TargetGroupInfo = [3, n0, _TGI, 0, [_n], [0]];
var TargetGroupPairInfo = [
    3,
    n0,
    _TGPI,
    0,
    [_tGa, _pTR, _tTR],
    [() => TargetGroupInfoList, () => TrafficRoute, () => TrafficRoute],
];
var TargetInstances = [
    3,
    n0,
    _TI,
    0,
    [_tFa, _aSG, _eTS],
    [() => EC2TagFilterList, 64 | 0, () => EC2TagSet],
];
var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException, ThrottlingException$1);
var TimeBasedCanary = [3, n0, _TBC, 0, [_cPa, _cIa], [1, 1]];
var TimeBasedLinear = [3, n0, _TBL, 0, [_lP, _lI], [1, 1]];
var TimeRange = [3, n0, _TR, 0, [_sta, _en], [4, 4]];
var TrafficRoute = [3, n0, _TRr, 0, [_lA], [64 | 0]];
var TrafficRoutingConfig = [
    3,
    n0,
    _TRC,
    0,
    [_ty, _tBC, _tBL],
    [0, () => TimeBasedCanary, () => TimeBasedLinear],
];
var TriggerConfig = [3, n0, _TC, 0, [_tNr, _tTA, _tE], [0, 0, 64 | 0]];
var TriggerTargetsLimitExceededException = [
    -3,
    n0,
    _TTLEE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TriggerTargetsLimitExceededException, TriggerTargetsLimitExceededException$1);
var UnsupportedActionForDeploymentTypeException = [
    -3,
    n0,
    _UAFDTE,
    {
        [_er]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(UnsupportedActionForDeploymentTypeException, UnsupportedActionForDeploymentTypeException$1);
var UntagResourceInput = [3, n0, _URI, 0, [_RA, _TK], [0, 64 | 0]];
var UntagResourceOutput = [3, n0, _URO, 0, [], []];
var UpdateApplicationInput = [3, n0, _UAI, 0, [_aN, _nAN], [0, 0]];
var UpdateDeploymentGroupInput = [
    3,
    n0,
    _UDGI,
    0,
    [_aN, _cDGN, _nDGN, _dCN, _eTF, _oPITF, _aSG, _sRA, _tC, _aC, _aRC, _oIS, _dS, _bGDC, _lBI, _eTS, _eS, _oPTS, _tHE],
    [
        0,
        0,
        0,
        0,
        () => EC2TagFilterList,
        () => TagFilterList,
        64 | 0,
        0,
        () => TriggerConfigList,
        () => AlarmConfiguration,
        () => AutoRollbackConfiguration,
        0,
        () => DeploymentStyle,
        () => BlueGreenDeploymentConfiguration,
        () => LoadBalancerInfo,
        () => EC2TagSet,
        () => ECSServiceList,
        () => OnPremisesTagSet,
        2,
    ],
];
var UpdateDeploymentGroupOutput = [
    3,
    n0,
    _UDGO,
    0,
    [_hNCU],
    [() => AutoScalingGroupList],
];
var ZonalConfig = [
    3,
    n0,
    _ZC,
    0,
    [_fZMDIS, _mDIS, _mHHPZ],
    [1, 1, () => MinimumHealthyHostsPerZone],
];
var __Unit = "unit";
var CodeDeployServiceException = [-3, _sm, "CodeDeployServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(CodeDeployServiceException, CodeDeployServiceException$1);
var AlarmList = [1, n0, _AL, 0, () => Alarm];
var ApplicationsInfoList = [1, n0, _AIL, 0, () => ApplicationInfo];
var AutoScalingGroupList = [1, n0, _ASGL, 0, () => AutoScalingGroup];
var DeploymentGroupInfoList = [1, n0, _DGIL, 0, () => DeploymentGroupInfo];
var DeploymentsInfoList = [1, n0, _DIL, 0, () => DeploymentInfo];
var DeploymentTargetList = [1, n0, _DTL, 0, () => DeploymentTarget];
var EC2TagFilterList = [1, n0, _ECTFL, 0, () => EC2TagFilter];
var EC2TagSetList = [1, n0, _ECTSL, 0, () => EC2TagFilterList];
var ECSServiceList = [1, n0, _ECSSL, 0, () => ECSService];
var ECSTaskSetList = [1, n0, _ECSTSL, 0, () => ECSTaskSet];
var ELBInfoList = [1, n0, _ELBIL, 0, () => ELBInfo];
var InstanceInfoList = [1, n0, _IIL, 0, () => InstanceInfo];
var InstanceSummaryList = [1, n0, _ISL, 0, () => InstanceSummary];
var LifecycleEventList = [1, n0, _LEL, 0, () => LifecycleEvent];
var OnPremisesTagSetList = [1, n0, _OPTSL, 0, () => TagFilterList];
var RevisionInfoList = [1, n0, _RIL, 0, () => RevisionInfo];
var RevisionLocationList = [1, n0, _RLL, 0, () => RevisionLocation];
var TagFilterList = [1, n0, _TFL, 0, () => TagFilter];
var TagList = [1, n0, _TL, 0, () => Tag];
var TargetGroupInfoList = [1, n0, _TGIL, 0, () => TargetGroupInfo];
var TargetGroupPairInfoList = [1, n0, _TGPIL, 0, () => TargetGroupPairInfo];
var TriggerConfigList = [1, n0, _TCL, 0, () => TriggerConfig];
var AddTagsToOnPremisesInstances = [
    9,
    n0,
    _ATTOPI,
    0,
    () => AddTagsToOnPremisesInstancesInput,
    () => __Unit,
];
var BatchGetApplicationRevisions = [
    9,
    n0,
    _BGAR,
    0,
    () => BatchGetApplicationRevisionsInput,
    () => BatchGetApplicationRevisionsOutput,
];
var BatchGetApplications = [
    9,
    n0,
    _BGA,
    0,
    () => BatchGetApplicationsInput,
    () => BatchGetApplicationsOutput,
];
var BatchGetDeploymentGroups = [
    9,
    n0,
    _BGDG,
    0,
    () => BatchGetDeploymentGroupsInput,
    () => BatchGetDeploymentGroupsOutput,
];
var BatchGetDeploymentInstances = [
    9,
    n0,
    _BGDIa,
    0,
    () => BatchGetDeploymentInstancesInput,
    () => BatchGetDeploymentInstancesOutput,
];
var BatchGetDeployments = [
    9,
    n0,
    _BGD,
    0,
    () => BatchGetDeploymentsInput,
    () => BatchGetDeploymentsOutput,
];
var BatchGetDeploymentTargets = [
    9,
    n0,
    _BGDT,
    0,
    () => BatchGetDeploymentTargetsInput,
    () => BatchGetDeploymentTargetsOutput,
];
var BatchGetOnPremisesInstances = [
    9,
    n0,
    _BGOPI,
    0,
    () => BatchGetOnPremisesInstancesInput,
    () => BatchGetOnPremisesInstancesOutput,
];
var ContinueDeployment = [9, n0, _CD, 0, () => ContinueDeploymentInput, () => __Unit];
var CreateApplication = [
    9,
    n0,
    _CA,
    0,
    () => CreateApplicationInput,
    () => CreateApplicationOutput,
];
var CreateDeployment = [
    9,
    n0,
    _CDr,
    0,
    () => CreateDeploymentInput,
    () => CreateDeploymentOutput,
];
var CreateDeploymentConfig = [
    9,
    n0,
    _CDC,
    0,
    () => CreateDeploymentConfigInput,
    () => CreateDeploymentConfigOutput,
];
var CreateDeploymentGroup = [
    9,
    n0,
    _CDG,
    0,
    () => CreateDeploymentGroupInput,
    () => CreateDeploymentGroupOutput,
];
var DeleteApplication = [9, n0, _DA, 0, () => DeleteApplicationInput, () => __Unit];
var DeleteDeploymentConfig = [
    9,
    n0,
    _DDC,
    0,
    () => DeleteDeploymentConfigInput,
    () => __Unit,
];
var DeleteDeploymentGroup = [
    9,
    n0,
    _DDG,
    0,
    () => DeleteDeploymentGroupInput,
    () => DeleteDeploymentGroupOutput,
];
var DeleteGitHubAccountToken = [
    9,
    n0,
    _DGHAT,
    0,
    () => DeleteGitHubAccountTokenInput,
    () => DeleteGitHubAccountTokenOutput,
];
var DeleteResourcesByExternalId = [
    9,
    n0,
    _DRBEI,
    0,
    () => DeleteResourcesByExternalIdInput,
    () => DeleteResourcesByExternalIdOutput,
];
var DeregisterOnPremisesInstance = [
    9,
    n0,
    _DOPI,
    0,
    () => DeregisterOnPremisesInstanceInput,
    () => __Unit,
];
var GetApplication = [
    9,
    n0,
    _GA,
    0,
    () => GetApplicationInput,
    () => GetApplicationOutput,
];
var GetApplicationRevision = [
    9,
    n0,
    _GAR,
    0,
    () => GetApplicationRevisionInput,
    () => GetApplicationRevisionOutput,
];
var GetDeployment = [9, n0, _GD, 0, () => GetDeploymentInput, () => GetDeploymentOutput];
var GetDeploymentConfig = [
    9,
    n0,
    _GDC,
    0,
    () => GetDeploymentConfigInput,
    () => GetDeploymentConfigOutput,
];
var GetDeploymentGroup = [
    9,
    n0,
    _GDG,
    0,
    () => GetDeploymentGroupInput,
    () => GetDeploymentGroupOutput,
];
var GetDeploymentInstance = [
    9,
    n0,
    _GDIe,
    0,
    () => GetDeploymentInstanceInput,
    () => GetDeploymentInstanceOutput,
];
var GetDeploymentTarget = [
    9,
    n0,
    _GDT,
    0,
    () => GetDeploymentTargetInput,
    () => GetDeploymentTargetOutput,
];
var GetOnPremisesInstance = [
    9,
    n0,
    _GOPI,
    0,
    () => GetOnPremisesInstanceInput,
    () => GetOnPremisesInstanceOutput,
];
var ListApplicationRevisions = [
    9,
    n0,
    _LAR,
    0,
    () => ListApplicationRevisionsInput,
    () => ListApplicationRevisionsOutput,
];
var ListApplications = [
    9,
    n0,
    _LA,
    0,
    () => ListApplicationsInput,
    () => ListApplicationsOutput,
];
var ListDeploymentConfigs = [
    9,
    n0,
    _LDC,
    0,
    () => ListDeploymentConfigsInput,
    () => ListDeploymentConfigsOutput,
];
var ListDeploymentGroups = [
    9,
    n0,
    _LDG,
    0,
    () => ListDeploymentGroupsInput,
    () => ListDeploymentGroupsOutput,
];
var ListDeploymentInstances = [
    9,
    n0,
    _LDIis,
    0,
    () => ListDeploymentInstancesInput,
    () => ListDeploymentInstancesOutput,
];
var ListDeployments = [
    9,
    n0,
    _LD,
    0,
    () => ListDeploymentsInput,
    () => ListDeploymentsOutput,
];
var ListDeploymentTargets = [
    9,
    n0,
    _LDT,
    0,
    () => ListDeploymentTargetsInput,
    () => ListDeploymentTargetsOutput,
];
var ListGitHubAccountTokenNames = [
    9,
    n0,
    _LGHATN,
    0,
    () => ListGitHubAccountTokenNamesInput,
    () => ListGitHubAccountTokenNamesOutput,
];
var ListOnPremisesInstances = [
    9,
    n0,
    _LOPI,
    0,
    () => ListOnPremisesInstancesInput,
    () => ListOnPremisesInstancesOutput,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    0,
    () => ListTagsForResourceInput,
    () => ListTagsForResourceOutput,
];
var PutLifecycleEventHookExecutionStatus = [
    9,
    n0,
    _PLEHES,
    0,
    () => PutLifecycleEventHookExecutionStatusInput,
    () => PutLifecycleEventHookExecutionStatusOutput,
];
var RegisterApplicationRevision = [
    9,
    n0,
    _RAR,
    0,
    () => RegisterApplicationRevisionInput,
    () => __Unit,
];
var RegisterOnPremisesInstance = [
    9,
    n0,
    _ROPI,
    0,
    () => RegisterOnPremisesInstanceInput,
    () => __Unit,
];
var RemoveTagsFromOnPremisesInstances = [
    9,
    n0,
    _RTFOPI,
    0,
    () => RemoveTagsFromOnPremisesInstancesInput,
    () => __Unit,
];
var SkipWaitTimeForInstanceTermination = [
    9,
    n0,
    _SWTFIT,
    0,
    () => SkipWaitTimeForInstanceTerminationInput,
    () => __Unit,
];
var StopDeployment = [
    9,
    n0,
    _SD,
    0,
    () => StopDeploymentInput,
    () => StopDeploymentOutput,
];
var TagResource = [9, n0, _TRa, 0, () => TagResourceInput, () => TagResourceOutput];
var UntagResource = [9, n0, _UR, 0, () => UntagResourceInput, () => UntagResourceOutput];
var UpdateApplication = [9, n0, _UA, 0, () => UpdateApplicationInput, () => __Unit];
var UpdateDeploymentGroup = [
    9,
    n0,
    _UDG,
    0,
    () => UpdateDeploymentGroupInput,
    () => UpdateDeploymentGroupOutput,
];

class AddTagsToOnPremisesInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "AddTagsToOnPremisesInstances", {})
    .n("CodeDeployClient", "AddTagsToOnPremisesInstancesCommand")
    .sc(AddTagsToOnPremisesInstances)
    .build() {
}

class BatchGetApplicationRevisionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetApplicationRevisions", {})
    .n("CodeDeployClient", "BatchGetApplicationRevisionsCommand")
    .sc(BatchGetApplicationRevisions)
    .build() {
}

class BatchGetApplicationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetApplications", {})
    .n("CodeDeployClient", "BatchGetApplicationsCommand")
    .sc(BatchGetApplications)
    .build() {
}

class BatchGetDeploymentGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetDeploymentGroups", {})
    .n("CodeDeployClient", "BatchGetDeploymentGroupsCommand")
    .sc(BatchGetDeploymentGroups)
    .build() {
}

class BatchGetDeploymentInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetDeploymentInstances", {})
    .n("CodeDeployClient", "BatchGetDeploymentInstancesCommand")
    .sc(BatchGetDeploymentInstances)
    .build() {
}

class BatchGetDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetDeployments", {})
    .n("CodeDeployClient", "BatchGetDeploymentsCommand")
    .sc(BatchGetDeployments)
    .build() {
}

class BatchGetDeploymentTargetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetDeploymentTargets", {})
    .n("CodeDeployClient", "BatchGetDeploymentTargetsCommand")
    .sc(BatchGetDeploymentTargets)
    .build() {
}

class BatchGetOnPremisesInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "BatchGetOnPremisesInstances", {})
    .n("CodeDeployClient", "BatchGetOnPremisesInstancesCommand")
    .sc(BatchGetOnPremisesInstances)
    .build() {
}

class ContinueDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ContinueDeployment", {})
    .n("CodeDeployClient", "ContinueDeploymentCommand")
    .sc(ContinueDeployment)
    .build() {
}

class CreateApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "CreateApplication", {})
    .n("CodeDeployClient", "CreateApplicationCommand")
    .sc(CreateApplication)
    .build() {
}

class CreateDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "CreateDeployment", {})
    .n("CodeDeployClient", "CreateDeploymentCommand")
    .sc(CreateDeployment)
    .build() {
}

class CreateDeploymentConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "CreateDeploymentConfig", {})
    .n("CodeDeployClient", "CreateDeploymentConfigCommand")
    .sc(CreateDeploymentConfig)
    .build() {
}

class CreateDeploymentGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "CreateDeploymentGroup", {})
    .n("CodeDeployClient", "CreateDeploymentGroupCommand")
    .sc(CreateDeploymentGroup)
    .build() {
}

class DeleteApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeleteApplication", {})
    .n("CodeDeployClient", "DeleteApplicationCommand")
    .sc(DeleteApplication)
    .build() {
}

class DeleteDeploymentConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeleteDeploymentConfig", {})
    .n("CodeDeployClient", "DeleteDeploymentConfigCommand")
    .sc(DeleteDeploymentConfig)
    .build() {
}

class DeleteDeploymentGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeleteDeploymentGroup", {})
    .n("CodeDeployClient", "DeleteDeploymentGroupCommand")
    .sc(DeleteDeploymentGroup)
    .build() {
}

class DeleteGitHubAccountTokenCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeleteGitHubAccountToken", {})
    .n("CodeDeployClient", "DeleteGitHubAccountTokenCommand")
    .sc(DeleteGitHubAccountToken)
    .build() {
}

class DeleteResourcesByExternalIdCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeleteResourcesByExternalId", {})
    .n("CodeDeployClient", "DeleteResourcesByExternalIdCommand")
    .sc(DeleteResourcesByExternalId)
    .build() {
}

class DeregisterOnPremisesInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "DeregisterOnPremisesInstance", {})
    .n("CodeDeployClient", "DeregisterOnPremisesInstanceCommand")
    .sc(DeregisterOnPremisesInstance)
    .build() {
}

class GetApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetApplication", {})
    .n("CodeDeployClient", "GetApplicationCommand")
    .sc(GetApplication)
    .build() {
}

class GetApplicationRevisionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetApplicationRevision", {})
    .n("CodeDeployClient", "GetApplicationRevisionCommand")
    .sc(GetApplicationRevision)
    .build() {
}

class GetDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetDeployment", {})
    .n("CodeDeployClient", "GetDeploymentCommand")
    .sc(GetDeployment)
    .build() {
}

class GetDeploymentConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetDeploymentConfig", {})
    .n("CodeDeployClient", "GetDeploymentConfigCommand")
    .sc(GetDeploymentConfig)
    .build() {
}

class GetDeploymentGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetDeploymentGroup", {})
    .n("CodeDeployClient", "GetDeploymentGroupCommand")
    .sc(GetDeploymentGroup)
    .build() {
}

class GetDeploymentInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetDeploymentInstance", {})
    .n("CodeDeployClient", "GetDeploymentInstanceCommand")
    .sc(GetDeploymentInstance)
    .build() {
}

class GetDeploymentTargetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetDeploymentTarget", {})
    .n("CodeDeployClient", "GetDeploymentTargetCommand")
    .sc(GetDeploymentTarget)
    .build() {
}

class GetOnPremisesInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "GetOnPremisesInstance", {})
    .n("CodeDeployClient", "GetOnPremisesInstanceCommand")
    .sc(GetOnPremisesInstance)
    .build() {
}

class ListApplicationRevisionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListApplicationRevisions", {})
    .n("CodeDeployClient", "ListApplicationRevisionsCommand")
    .sc(ListApplicationRevisions)
    .build() {
}

class ListApplicationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListApplications", {})
    .n("CodeDeployClient", "ListApplicationsCommand")
    .sc(ListApplications)
    .build() {
}

class ListDeploymentConfigsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListDeploymentConfigs", {})
    .n("CodeDeployClient", "ListDeploymentConfigsCommand")
    .sc(ListDeploymentConfigs)
    .build() {
}

class ListDeploymentGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListDeploymentGroups", {})
    .n("CodeDeployClient", "ListDeploymentGroupsCommand")
    .sc(ListDeploymentGroups)
    .build() {
}

class ListDeploymentInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListDeploymentInstances", {})
    .n("CodeDeployClient", "ListDeploymentInstancesCommand")
    .sc(ListDeploymentInstances)
    .build() {
}

class ListDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListDeployments", {})
    .n("CodeDeployClient", "ListDeploymentsCommand")
    .sc(ListDeployments)
    .build() {
}

class ListDeploymentTargetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListDeploymentTargets", {})
    .n("CodeDeployClient", "ListDeploymentTargetsCommand")
    .sc(ListDeploymentTargets)
    .build() {
}

class ListGitHubAccountTokenNamesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListGitHubAccountTokenNames", {})
    .n("CodeDeployClient", "ListGitHubAccountTokenNamesCommand")
    .sc(ListGitHubAccountTokenNames)
    .build() {
}

class ListOnPremisesInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListOnPremisesInstances", {})
    .n("CodeDeployClient", "ListOnPremisesInstancesCommand")
    .sc(ListOnPremisesInstances)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "ListTagsForResource", {})
    .n("CodeDeployClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class PutLifecycleEventHookExecutionStatusCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "PutLifecycleEventHookExecutionStatus", {})
    .n("CodeDeployClient", "PutLifecycleEventHookExecutionStatusCommand")
    .sc(PutLifecycleEventHookExecutionStatus)
    .build() {
}

class RegisterApplicationRevisionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "RegisterApplicationRevision", {})
    .n("CodeDeployClient", "RegisterApplicationRevisionCommand")
    .sc(RegisterApplicationRevision)
    .build() {
}

class RegisterOnPremisesInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "RegisterOnPremisesInstance", {})
    .n("CodeDeployClient", "RegisterOnPremisesInstanceCommand")
    .sc(RegisterOnPremisesInstance)
    .build() {
}

class RemoveTagsFromOnPremisesInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "RemoveTagsFromOnPremisesInstances", {})
    .n("CodeDeployClient", "RemoveTagsFromOnPremisesInstancesCommand")
    .sc(RemoveTagsFromOnPremisesInstances)
    .build() {
}

class SkipWaitTimeForInstanceTerminationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "SkipWaitTimeForInstanceTermination", {})
    .n("CodeDeployClient", "SkipWaitTimeForInstanceTerminationCommand")
    .sc(SkipWaitTimeForInstanceTermination)
    .build() {
}

class StopDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "StopDeployment", {})
    .n("CodeDeployClient", "StopDeploymentCommand")
    .sc(StopDeployment)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "TagResource", {})
    .n("CodeDeployClient", "TagResourceCommand")
    .sc(TagResource)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "UntagResource", {})
    .n("CodeDeployClient", "UntagResourceCommand")
    .sc(UntagResource)
    .build() {
}

class UpdateApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "UpdateApplication", {})
    .n("CodeDeployClient", "UpdateApplicationCommand")
    .sc(UpdateApplication)
    .build() {
}

class UpdateDeploymentGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeDeploy_20141006", "UpdateDeploymentGroup", {})
    .n("CodeDeployClient", "UpdateDeploymentGroupCommand")
    .sc(UpdateDeploymentGroup)
    .build() {
}

const commands = {
    AddTagsToOnPremisesInstancesCommand,
    BatchGetApplicationRevisionsCommand,
    BatchGetApplicationsCommand,
    BatchGetDeploymentGroupsCommand,
    BatchGetDeploymentInstancesCommand,
    BatchGetDeploymentsCommand,
    BatchGetDeploymentTargetsCommand,
    BatchGetOnPremisesInstancesCommand,
    ContinueDeploymentCommand,
    CreateApplicationCommand,
    CreateDeploymentCommand,
    CreateDeploymentConfigCommand,
    CreateDeploymentGroupCommand,
    DeleteApplicationCommand,
    DeleteDeploymentConfigCommand,
    DeleteDeploymentGroupCommand,
    DeleteGitHubAccountTokenCommand,
    DeleteResourcesByExternalIdCommand,
    DeregisterOnPremisesInstanceCommand,
    GetApplicationCommand,
    GetApplicationRevisionCommand,
    GetDeploymentCommand,
    GetDeploymentConfigCommand,
    GetDeploymentGroupCommand,
    GetDeploymentInstanceCommand,
    GetDeploymentTargetCommand,
    GetOnPremisesInstanceCommand,
    ListApplicationRevisionsCommand,
    ListApplicationsCommand,
    ListDeploymentConfigsCommand,
    ListDeploymentGroupsCommand,
    ListDeploymentInstancesCommand,
    ListDeploymentsCommand,
    ListDeploymentTargetsCommand,
    ListGitHubAccountTokenNamesCommand,
    ListOnPremisesInstancesCommand,
    ListTagsForResourceCommand,
    PutLifecycleEventHookExecutionStatusCommand,
    RegisterApplicationRevisionCommand,
    RegisterOnPremisesInstanceCommand,
    RemoveTagsFromOnPremisesInstancesCommand,
    SkipWaitTimeForInstanceTerminationCommand,
    StopDeploymentCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateApplicationCommand,
    UpdateDeploymentGroupCommand,
};
class CodeDeploy extends CodeDeployClient {
}
smithyClient.createAggregatedClient(commands, CodeDeploy);

const paginateListApplicationRevisions = core.createPaginator(CodeDeployClient, ListApplicationRevisionsCommand, "nextToken", "nextToken", "");

const paginateListApplications = core.createPaginator(CodeDeployClient, ListApplicationsCommand, "nextToken", "nextToken", "");

const paginateListDeploymentConfigs = core.createPaginator(CodeDeployClient, ListDeploymentConfigsCommand, "nextToken", "nextToken", "");

const paginateListDeploymentGroups = core.createPaginator(CodeDeployClient, ListDeploymentGroupsCommand, "nextToken", "nextToken", "");

const paginateListDeploymentInstances = core.createPaginator(CodeDeployClient, ListDeploymentInstancesCommand, "nextToken", "nextToken", "");

const paginateListDeployments = core.createPaginator(CodeDeployClient, ListDeploymentsCommand, "nextToken", "nextToken", "");

const checkState = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new GetDeploymentCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.deploymentInfo.status;
            };
            if (returnComparator() === "Succeeded") {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.deploymentInfo.status;
            };
            if (returnComparator() === "Failed") {
                return { state: utilWaiter.WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.deploymentInfo.status;
            };
            if (returnComparator() === "Stopped") {
                return { state: utilWaiter.WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDeploymentSuccessful = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilDeploymentSuccessful = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return utilWaiter.checkExceptions(result);
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AddTagsToOnPremisesInstancesCommand = AddTagsToOnPremisesInstancesCommand;
exports.AlarmsLimitExceededException = AlarmsLimitExceededException$1;
exports.ApplicationAlreadyExistsException = ApplicationAlreadyExistsException$1;
exports.ApplicationDoesNotExistException = ApplicationDoesNotExistException$1;
exports.ApplicationLimitExceededException = ApplicationLimitExceededException$1;
exports.ApplicationNameRequiredException = ApplicationNameRequiredException$1;
exports.ApplicationRevisionSortBy = ApplicationRevisionSortBy;
exports.ArnNotSupportedException = ArnNotSupportedException$1;
exports.AutoRollbackEvent = AutoRollbackEvent;
exports.BatchGetApplicationRevisionsCommand = BatchGetApplicationRevisionsCommand;
exports.BatchGetApplicationsCommand = BatchGetApplicationsCommand;
exports.BatchGetDeploymentGroupsCommand = BatchGetDeploymentGroupsCommand;
exports.BatchGetDeploymentInstancesCommand = BatchGetDeploymentInstancesCommand;
exports.BatchGetDeploymentTargetsCommand = BatchGetDeploymentTargetsCommand;
exports.BatchGetDeploymentsCommand = BatchGetDeploymentsCommand;
exports.BatchGetOnPremisesInstancesCommand = BatchGetOnPremisesInstancesCommand;
exports.BatchLimitExceededException = BatchLimitExceededException$1;
exports.BucketNameFilterRequiredException = BucketNameFilterRequiredException$1;
exports.BundleType = BundleType;
exports.CodeDeploy = CodeDeploy;
exports.CodeDeployClient = CodeDeployClient;
exports.CodeDeployServiceException = CodeDeployServiceException$1;
exports.ComputePlatform = ComputePlatform;
exports.ContinueDeploymentCommand = ContinueDeploymentCommand;
exports.CreateApplicationCommand = CreateApplicationCommand;
exports.CreateDeploymentCommand = CreateDeploymentCommand;
exports.CreateDeploymentConfigCommand = CreateDeploymentConfigCommand;
exports.CreateDeploymentGroupCommand = CreateDeploymentGroupCommand;
exports.DeleteApplicationCommand = DeleteApplicationCommand;
exports.DeleteDeploymentConfigCommand = DeleteDeploymentConfigCommand;
exports.DeleteDeploymentGroupCommand = DeleteDeploymentGroupCommand;
exports.DeleteGitHubAccountTokenCommand = DeleteGitHubAccountTokenCommand;
exports.DeleteResourcesByExternalIdCommand = DeleteResourcesByExternalIdCommand;
exports.DeploymentAlreadyCompletedException = DeploymentAlreadyCompletedException$1;
exports.DeploymentConfigAlreadyExistsException = DeploymentConfigAlreadyExistsException$1;
exports.DeploymentConfigDoesNotExistException = DeploymentConfigDoesNotExistException$1;
exports.DeploymentConfigInUseException = DeploymentConfigInUseException$1;
exports.DeploymentConfigLimitExceededException = DeploymentConfigLimitExceededException$1;
exports.DeploymentConfigNameRequiredException = DeploymentConfigNameRequiredException$1;
exports.DeploymentCreator = DeploymentCreator;
exports.DeploymentDoesNotExistException = DeploymentDoesNotExistException$1;
exports.DeploymentGroupAlreadyExistsException = DeploymentGroupAlreadyExistsException$1;
exports.DeploymentGroupDoesNotExistException = DeploymentGroupDoesNotExistException$1;
exports.DeploymentGroupLimitExceededException = DeploymentGroupLimitExceededException$1;
exports.DeploymentGroupNameRequiredException = DeploymentGroupNameRequiredException$1;
exports.DeploymentIdRequiredException = DeploymentIdRequiredException$1;
exports.DeploymentIsNotInReadyStateException = DeploymentIsNotInReadyStateException$1;
exports.DeploymentLimitExceededException = DeploymentLimitExceededException$1;
exports.DeploymentNotStartedException = DeploymentNotStartedException$1;
exports.DeploymentOption = DeploymentOption;
exports.DeploymentReadyAction = DeploymentReadyAction;
exports.DeploymentStatus = DeploymentStatus;
exports.DeploymentTargetDoesNotExistException = DeploymentTargetDoesNotExistException$1;
exports.DeploymentTargetIdRequiredException = DeploymentTargetIdRequiredException$1;
exports.DeploymentTargetListSizeExceededException = DeploymentTargetListSizeExceededException$1;
exports.DeploymentTargetType = DeploymentTargetType;
exports.DeploymentType = DeploymentType;
exports.DeploymentWaitType = DeploymentWaitType;
exports.DeregisterOnPremisesInstanceCommand = DeregisterOnPremisesInstanceCommand;
exports.DescriptionTooLongException = DescriptionTooLongException$1;
exports.EC2TagFilterType = EC2TagFilterType;
exports.ECSServiceMappingLimitExceededException = ECSServiceMappingLimitExceededException$1;
exports.ErrorCode = ErrorCode;
exports.FileExistsBehavior = FileExistsBehavior;
exports.GetApplicationCommand = GetApplicationCommand;
exports.GetApplicationRevisionCommand = GetApplicationRevisionCommand;
exports.GetDeploymentCommand = GetDeploymentCommand;
exports.GetDeploymentConfigCommand = GetDeploymentConfigCommand;
exports.GetDeploymentGroupCommand = GetDeploymentGroupCommand;
exports.GetDeploymentInstanceCommand = GetDeploymentInstanceCommand;
exports.GetDeploymentTargetCommand = GetDeploymentTargetCommand;
exports.GetOnPremisesInstanceCommand = GetOnPremisesInstanceCommand;
exports.GitHubAccountTokenDoesNotExistException = GitHubAccountTokenDoesNotExistException$1;
exports.GitHubAccountTokenNameRequiredException = GitHubAccountTokenNameRequiredException$1;
exports.GreenFleetProvisioningAction = GreenFleetProvisioningAction;
exports.IamArnRequiredException = IamArnRequiredException$1;
exports.IamSessionArnAlreadyRegisteredException = IamSessionArnAlreadyRegisteredException$1;
exports.IamUserArnAlreadyRegisteredException = IamUserArnAlreadyRegisteredException$1;
exports.IamUserArnRequiredException = IamUserArnRequiredException$1;
exports.InstanceAction = InstanceAction;
exports.InstanceDoesNotExistException = InstanceDoesNotExistException$1;
exports.InstanceIdRequiredException = InstanceIdRequiredException$1;
exports.InstanceLimitExceededException = InstanceLimitExceededException$1;
exports.InstanceNameAlreadyRegisteredException = InstanceNameAlreadyRegisteredException$1;
exports.InstanceNameRequiredException = InstanceNameRequiredException$1;
exports.InstanceNotRegisteredException = InstanceNotRegisteredException$1;
exports.InstanceStatus = InstanceStatus;
exports.InvalidAlarmConfigException = InvalidAlarmConfigException$1;
exports.InvalidApplicationNameException = InvalidApplicationNameException$1;
exports.InvalidArnException = InvalidArnException$1;
exports.InvalidAutoRollbackConfigException = InvalidAutoRollbackConfigException$1;
exports.InvalidAutoScalingGroupException = InvalidAutoScalingGroupException$1;
exports.InvalidBlueGreenDeploymentConfigurationException = InvalidBlueGreenDeploymentConfigurationException$1;
exports.InvalidBucketNameFilterException = InvalidBucketNameFilterException$1;
exports.InvalidComputePlatformException = InvalidComputePlatformException$1;
exports.InvalidDeployedStateFilterException = InvalidDeployedStateFilterException$1;
exports.InvalidDeploymentConfigNameException = InvalidDeploymentConfigNameException$1;
exports.InvalidDeploymentGroupNameException = InvalidDeploymentGroupNameException$1;
exports.InvalidDeploymentIdException = InvalidDeploymentIdException$1;
exports.InvalidDeploymentInstanceTypeException = InvalidDeploymentInstanceTypeException$1;
exports.InvalidDeploymentStatusException = InvalidDeploymentStatusException$1;
exports.InvalidDeploymentStyleException = InvalidDeploymentStyleException$1;
exports.InvalidDeploymentTargetIdException = InvalidDeploymentTargetIdException$1;
exports.InvalidDeploymentWaitTypeException = InvalidDeploymentWaitTypeException$1;
exports.InvalidEC2TagCombinationException = InvalidEC2TagCombinationException$1;
exports.InvalidEC2TagException = InvalidEC2TagException$1;
exports.InvalidECSServiceException = InvalidECSServiceException$1;
exports.InvalidExternalIdException = InvalidExternalIdException$1;
exports.InvalidFileExistsBehaviorException = InvalidFileExistsBehaviorException$1;
exports.InvalidGitHubAccountTokenException = InvalidGitHubAccountTokenException$1;
exports.InvalidGitHubAccountTokenNameException = InvalidGitHubAccountTokenNameException$1;
exports.InvalidIamSessionArnException = InvalidIamSessionArnException$1;
exports.InvalidIamUserArnException = InvalidIamUserArnException$1;
exports.InvalidIgnoreApplicationStopFailuresValueException = InvalidIgnoreApplicationStopFailuresValueException$1;
exports.InvalidInputException = InvalidInputException$1;
exports.InvalidInstanceNameException = InvalidInstanceNameException$1;
exports.InvalidInstanceStatusException = InvalidInstanceStatusException$1;
exports.InvalidInstanceTypeException = InvalidInstanceTypeException$1;
exports.InvalidKeyPrefixFilterException = InvalidKeyPrefixFilterException$1;
exports.InvalidLifecycleEventHookExecutionIdException = InvalidLifecycleEventHookExecutionIdException$1;
exports.InvalidLifecycleEventHookExecutionStatusException = InvalidLifecycleEventHookExecutionStatusException$1;
exports.InvalidLoadBalancerInfoException = InvalidLoadBalancerInfoException$1;
exports.InvalidMinimumHealthyHostValueException = InvalidMinimumHealthyHostValueException$1;
exports.InvalidNextTokenException = InvalidNextTokenException$1;
exports.InvalidOnPremisesTagCombinationException = InvalidOnPremisesTagCombinationException$1;
exports.InvalidOperationException = InvalidOperationException$1;
exports.InvalidRegistrationStatusException = InvalidRegistrationStatusException$1;
exports.InvalidRevisionException = InvalidRevisionException$1;
exports.InvalidRoleException = InvalidRoleException$1;
exports.InvalidSortByException = InvalidSortByException$1;
exports.InvalidSortOrderException = InvalidSortOrderException$1;
exports.InvalidTagException = InvalidTagException$1;
exports.InvalidTagFilterException = InvalidTagFilterException$1;
exports.InvalidTagsToAddException = InvalidTagsToAddException$1;
exports.InvalidTargetFilterNameException = InvalidTargetFilterNameException$1;
exports.InvalidTargetGroupPairException = InvalidTargetGroupPairException$1;
exports.InvalidTargetInstancesException = InvalidTargetInstancesException$1;
exports.InvalidTimeRangeException = InvalidTimeRangeException$1;
exports.InvalidTrafficRoutingConfigurationException = InvalidTrafficRoutingConfigurationException$1;
exports.InvalidTriggerConfigException = InvalidTriggerConfigException$1;
exports.InvalidUpdateOutdatedInstancesOnlyValueException = InvalidUpdateOutdatedInstancesOnlyValueException$1;
exports.InvalidZonalDeploymentConfigurationException = InvalidZonalDeploymentConfigurationException$1;
exports.LifecycleErrorCode = LifecycleErrorCode;
exports.LifecycleEventAlreadyCompletedException = LifecycleEventAlreadyCompletedException$1;
exports.LifecycleEventStatus = LifecycleEventStatus;
exports.LifecycleHookLimitExceededException = LifecycleHookLimitExceededException$1;
exports.ListApplicationRevisionsCommand = ListApplicationRevisionsCommand;
exports.ListApplicationsCommand = ListApplicationsCommand;
exports.ListDeploymentConfigsCommand = ListDeploymentConfigsCommand;
exports.ListDeploymentGroupsCommand = ListDeploymentGroupsCommand;
exports.ListDeploymentInstancesCommand = ListDeploymentInstancesCommand;
exports.ListDeploymentTargetsCommand = ListDeploymentTargetsCommand;
exports.ListDeploymentsCommand = ListDeploymentsCommand;
exports.ListGitHubAccountTokenNamesCommand = ListGitHubAccountTokenNamesCommand;
exports.ListOnPremisesInstancesCommand = ListOnPremisesInstancesCommand;
exports.ListStateFilterAction = ListStateFilterAction;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.MinimumHealthyHostsPerZoneType = MinimumHealthyHostsPerZoneType;
exports.MinimumHealthyHostsType = MinimumHealthyHostsType;
exports.MultipleIamArnsProvidedException = MultipleIamArnsProvidedException$1;
exports.OperationNotSupportedException = OperationNotSupportedException$1;
exports.OutdatedInstancesStrategy = OutdatedInstancesStrategy;
exports.PutLifecycleEventHookExecutionStatusCommand = PutLifecycleEventHookExecutionStatusCommand;
exports.RegisterApplicationRevisionCommand = RegisterApplicationRevisionCommand;
exports.RegisterOnPremisesInstanceCommand = RegisterOnPremisesInstanceCommand;
exports.RegistrationStatus = RegistrationStatus;
exports.RemoveTagsFromOnPremisesInstancesCommand = RemoveTagsFromOnPremisesInstancesCommand;
exports.ResourceArnRequiredException = ResourceArnRequiredException$1;
exports.ResourceValidationException = ResourceValidationException$1;
exports.RevisionDoesNotExistException = RevisionDoesNotExistException$1;
exports.RevisionLocationType = RevisionLocationType;
exports.RevisionRequiredException = RevisionRequiredException$1;
exports.RoleRequiredException = RoleRequiredException$1;
exports.SkipWaitTimeForInstanceTerminationCommand = SkipWaitTimeForInstanceTerminationCommand;
exports.SortOrder = SortOrder;
exports.StopDeploymentCommand = StopDeploymentCommand;
exports.StopStatus = StopStatus;
exports.TagFilterType = TagFilterType;
exports.TagLimitExceededException = TagLimitExceededException$1;
exports.TagRequiredException = TagRequiredException$1;
exports.TagResourceCommand = TagResourceCommand;
exports.TagSetListLimitExceededException = TagSetListLimitExceededException$1;
exports.TargetFilterName = TargetFilterName;
exports.TargetLabel = TargetLabel;
exports.TargetStatus = TargetStatus;
exports.ThrottlingException = ThrottlingException$1;
exports.TrafficRoutingType = TrafficRoutingType;
exports.TriggerEventType = TriggerEventType;
exports.TriggerTargetsLimitExceededException = TriggerTargetsLimitExceededException$1;
exports.UnsupportedActionForDeploymentTypeException = UnsupportedActionForDeploymentTypeException$1;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateApplicationCommand = UpdateApplicationCommand;
exports.UpdateDeploymentGroupCommand = UpdateDeploymentGroupCommand;
exports._InstanceType = _InstanceType;
exports.paginateListApplicationRevisions = paginateListApplicationRevisions;
exports.paginateListApplications = paginateListApplications;
exports.paginateListDeploymentConfigs = paginateListDeploymentConfigs;
exports.paginateListDeploymentGroups = paginateListDeploymentGroups;
exports.paginateListDeploymentInstances = paginateListDeploymentInstances;
exports.paginateListDeployments = paginateListDeployments;
exports.waitForDeploymentSuccessful = waitForDeploymentSuccessful;
exports.waitUntilDeploymentSuccessful = waitUntilDeploymentSuccessful;
